
## Deciphering the Characteristics of Macrophages and Gene Regulatory Mechanisms in Pancreatic Cancer through Integrating Single - Cell and Spatial Transcriptomics

#### Abstract
Many studies have shown that pancreatic cancer is one of the cancers with extremely high mortality. The poor prognosis and lack of early diagnostic methods remain major challenges in the treatment of this cancer. In this article, in order to better detect the occurrence of pancreatic cancer, tools such as monocle3, singleR, harmony in R, and scanpy in Python were used to analyze the cells and genes of pancreatic cancer tissues in mice. By analyzing the data, the following results were obtained:

#### Introduction
Pancreatic ductal adenocarcinoma (PDAC) is the most common type of pancreatic cancer<sup>[1](#ref1)</sup> and ranks seventh among the leading causes of death due to cancer<sup>[2](#ref2)</sup>. Pancreatic cancer is increasing dramatically worldwide and is expected to continue to be the leading cause of cancer-related death<sup>[3](#ref3)</sup>. Due to the lack of specific symptoms and efficient and convenient examination methods, the early diagnosis of PDAC is extremely difficult<sup>[4](#ref4)</sup>. Therefore, most cases are diagnosed in a late stage that is ineffective to be treated by current therapies<sup>[5](#ref5)</sup>, and the prognosis of PDAC is extremely poor. It is important to study pancreatic cancer in order to better treat it.

Macrophages are a diverse group of immune system cells present in all body sites<sup>[6](#ref6)</sup> and are the major component of leukocyte infiltration<sup>[7](#ref7)</sup>. Macrophages exhibit a high degree of plasticity in response to various environmental stimuli<sup>[10](#ref10)</sup>. Under pathological conditions, evidence of local proliferation and recruitment exists, and differences are observed according to tissue location and type of inflammatory damage<sup>[11](#ref11)</sup>. Tumors can distort the original function of macrophages<sup>[12](#ref12)</sup> and form tumor-associated macrophages (TAMs), which stimulate cell proliferation and angiogenesis, thereby supporting tumor growth and metastasis<sup>[13](#ref13)</sup>. Macrophages are involved in the formation of tumor microenvironment. The tumor microenvironment (TME) is a dynamic network composed of malignant cells, immune cells and other components<sup>[14](#ref14)</sup>. The tumor microenvironment can actively alter the metabolism of macrophages that determine the macrophage phenotype, either through direct metabolite exchange or through cytokines and other signaling mediators<sup>[15](#ref15)</sup>. So, in the tumor microenvironment, dense interactions with tumor cells, immune cells, and stromal cells can drive either pro-tumor or antineoplastic macrophage phenotypes. TAMs<sup>[14](#ref14)</sup> in turn strongly shape the levels of cytokines and metabolites in the tumor microenvironment and thus play a central role in antitumor immunity<sup>[16](#ref16)</sup>. There is substantial clinical and experimental evidence that macrophages contribute to tumorigenesis and malignant progression<sup>[6](#ref6)</sup>. TAMs are the main component of tumor microenvironment, which is closely related to tumor metastasis<sup>[17](#ref17)</sup>. TAMs are key drivers of tumor progression, metastasis, and treatment resistance<sup>[18](#ref18)</sup>. Moreover, many properties of metastatic cancer cells are also seen in normal macrophages<sup>[19](#ref19)</sup>. TAMs are considered to be the main immunomodulatory cells in the TME of the tumor microenvironment, which can directly inhibit the activation of T lymphocytes and natural killer (NK) cells, thereby impairing the protective immune response<sup>[20](#ref20)</sup>. Oncogenic transformation of the most prevalent epithelial cell line leads to maladaptation of the resident tissue macrophage differentiation pathway, producing parenchymal and stromal TAMs that largely contribute to cancer progression<sup>[21](#ref21)</sup>. Depending on the microenvironment and various cues, macrophages are polarized to either an M1-like pro-inflammatory phenotype or the anti-inflammatory M2-like phenotype<sup>[10](#ref10)</sup>. Signal-regulatory protein alpha (SIRPα) on macrophages connects to CD47, a "don't eat me" signal on cancer cells that prevents the macrophages from swallowing cancer cells. Colony-stimulating factors secreted by cancer cells polarize TAMs into tumor-inducing M2 phenotypes<sup>[22](#ref22)</sup>. TAMs mainly play an M2-like tumor promoting role in TME<sup>[23](#ref23)</sup> and produce growth-promoting molecules that actively stimulate tumor growth<sup>[24](#ref24)</sup>. They are also capable of delaying antitumor immune responses by synthesizing immunosuppressive cytokines, expressing T cell co-inhibitory molecules, and depleting amino acids critical for effector T cell activity<sup>[25](#ref25)</sup>. Among them, M2d macrophages, also known as TAMs, can be activated by TLR ligands and A2 adenosine receptor agonists and play an important role in regulating tumor progression, angiogenesis and metastasis<sup>[26](#ref26)</sup>. PDAC is characterized by an extensive fibrous inflammatory stroma<sup>[27](#ref27)</sup>, including a large number of infiltrating heterogeneous immune-cell populations<sup>[28](#ref28)</sup>, tumor-associated fibroblasts (CAFs), extracellular matrix (ECM) proteins, and stellate cells<sup>[29](#ref29)</sup>. TAMs, one of the most abundant immune cell populations in the stroma of pancreatic tumors<sup>[30](#ref30)</sup>, are ubiquitous in the stroma and a key driver of immunosuppression<sup>[31](#ref31)</sup>. From the initiation of carcinogenesis, the immune response to pancreatic cancer leads to an immunosuppressed tumor microenvironment (TME)<sup>[32](#ref32)</sup>. Oncostatin M (OSM) secreted by macrophages stimulates the expression of inflammatory genes in CAFs, which subsequently induces a pro-tumor environment and participates in tumor cell survival and migration signaling pathways<sup>[33](#ref33)</sup>. In the PDAC microenvironment, the infiltration of inflammatory cells is imbalanced to an immunosuppressive phenotype, and the proportion of myeloid derived suppressor cells (MDSC), M2-polarized macrophages and Tregs is higher than that of M1 macrophages, dendritic cells and effector CD4+ and CD8+ T lymphocytes<sup>[34](#ref34)</sup>. M2 macrophages can secrete vascular endothelial growth factor (VEGF), which promotes angiogenesis<sup>[35](#ref35)</sup>, and they are positively correlated with microvessel density (MVD) in PDAC tissues<sup>[36](#ref36)</sup>.

Single-cell sequencing refers to the sequencing technology to obtain the genetic information of a single cell, that is, the extraction, amplification and high-throughput sequencing analysis of the genome or transcriptome at the level of a single cell<sup>[37](#ref37)</sup>. In cancer research, single-cell sequencing is an important method for studying applications such as cell line development or single-cell analysis<sup>[37](#ref37)</sup>. Studies using single-cell rna sequencing (scRNA-seq) and spatial transcriptomics can improve the understanding of the ontogeny, phenotypic, and functional plasticity of macrophages<sup>[38](#ref38)</sup>. It can also be used to infer macrophage tumor interaction networks, while pseudo-time trajectories can be used to dissect cell evolution and dynamics<sup>[39](#ref39)</sup>.

Spatial transcriptomics will use smFISH to generate mRNA. The released mRNA moves and attaches to the matrix below the tissue, from which the location of the mRNA in the tissue is known. Complementary cDNA molecules were obtained after reverse transcription. Imaging was performed with cDNA<sup>[40](#ref40)</sup>. 

In the study of cancer, it can be used to study the heterogeneity of tumors, analyze the gene expression patterns of different cell types in tumor tissues, reveal the interaction between tumor cells and the surrounding microenvironment, and the molecular mechanism of tumor occurrence and development. An example is the analysis of Macrophages infiltrating tissues<sup>[41](#ref41)</sup>. Using spatial transcriptomics, the position of cells in tissues can be understood and the results of quantitative measurement of gene expression can be obtained<sup>[40](#ref40)</sup>.

#### Result
**1.** To elucidate cell type-specific clustering patterns across four distinct sample categories, we implemented Harmony for batch effect correction and dataset integration. Principal Component Analysis (PCA) was applied for dimensionality reduction, followed by UMAP visualization to resolve intrinsic clustering architectures. This pipeline effectively preserved biological heterogeneity while mitigating technical variability across samples. Comparative analysis revealed that T-cells in healthy peripheral blood mononuclear cells (PBMCs) exhibited broader spatial dispersion relative to pancreatic ductal adenocarcinoma (PDAC) tissues ([Fig.1 C, F](#figure2)). Conversely, macrophages in PDAC tissues displayed expanded distribution compared to adjacent non-tumor regions ([Fig.1 K, L](#figure2)).
<a id="fig1"></a>
![fig1](images/1.JPG)

**Fig. 1. UMAP plots and harmony-integrated for PDAC tissue and adjacent normal tissue.**   
**(A)** Unintegrated UMAP plot for PDAC PBMCs grouped by sample name                                                        
**(B)** Integrated UMAP plot for PDAC PBMCs grouped by sample name   
**(C)** Unintegrated UMAP plot for adjacent normal PBMCs grouped by sample name
**(D)** Unintegrated UMAP plot for healthy PBMCs grouped by sample name                                                          
**(E)** Integrated UMAP plot for healthy PBMCs  grouped by sample name 
**(F)** Integrated UMAP plot for healthy PBMCs  grouped by sample name 
**(G)** Unintegrated UMAP plot for PDAC PBMCs grouped by sample name                                                        
**(H)** Integrated UMAP plot for PDAC PBMCs grouped by sample name   
**(I)** Unintegrated UMAP plot for adjacent normal PBMCs grouped by sample name
**(J)** Unintegrated UMAP plot for healthy PBMCs grouped by sample name                                                          
**(K)** Integrated UMAP plot for healthy PBMCs  grouped by sample name 
**(L)** Integrated UMAP plot for healthy PBMCs  grouped by sample name 


 


**2.** To profile transcriptional heterogeneity across cell types, we generated bubble plots with cell types annotated on the x-axis and gene symbols on the y-axis. Bubble size quantifies expression magnitude, while color gradients denote detection frequency, enabling simultaneous visualization of quantitative and qualitative expression features. Notably, the top 10 highly expressed genes in healthy PBMCs demonstrated marked divergence from PDAC PBMCs ([Fig.2 A, B](#fig2)). Tissue-specific markers were identified: LCN2 dominated in healthy tissues, whereas CTRB1 was upregulated in PDAC. Adjacent normal tissues showed preferential expression of SDPR, while AMY2A was enriched in PDAC regions. 

<a id="fig2"></a>
![fig2](images/2.JPG)

**Fig. 2. UMAP plots and harmony-integrated for PDAC tissue and adjacent normal tissue.** 
**(A)** Unintegrated UMAP plot for PDAC PBMCs grouped by sample name                                                        
**(B)** Integrated UMAP plot for PDAC PBMCs grouped by sample name   
**(C)** Unintegrated UMAP plot for adjacent normal PBMCs grouped by sample name
**(D)** Unintegrated UMAP plot for healthy PBMCs grouped by sample name                                                          
**(E)** Integrated UMAP plot for healthy PBMCs  grouped by sample name 
**(F)** Integrated UMAP plot for healthy PBMCs  grouped by sample name 

The figures labeled as a, b, and c show the unintegrated UMAP plots for the first four samples of PDAC tissue,harmony-integrated plots that are categorized by sample classification and by cell type classification.The figures labeled as d, e, and f show the unintegrated UMAP plots for the first four samples of adjacent normal tissue,harmony-integrated plots that are categorized by sample classification and by cell type classification.



**3.** To investigate the gene expression dynamics within each cell type, we employed pseudotime analysis. By ordering cells along a pseudotemporal continuum based on transcriptional similarity, we delineated transitional states and lineage commitment patterns. Integrated trajectory analysis revealed that common myeloid progenitors (CMPs) bifurcated into monocytes and macrophages during development ([Fig.4](#fig4)).
<a id="fig4"></a>
![fig4](images/4.JPG)

**Fig. 5. Pseudotime analysis for macrophage in PDAC PBMC, healthy PBMC, PDAC tissue, and adjacent normal tissuePBMCs.** 
**(A)**  Bubble charts for expression levels of the top ten genes in PDAC tissue PBMC                                                     
**(B)** Bubble charts for expression levels of the top ten genes in adjacent normal tissue

Figure a displays the expression levels of the top ten genes with the highest expression in each cell within the integrated PDAC tissue samples. Figure b shows the expression levels of the top ten genes with the highest expression in each cell within the integrated adjacent normal tissue samples.

 

**3.**  Pseudotime analysis using single-cell RNA sequencing (scRNA-seq) data reconstructed macrophage differentiation dynamics. By ordering cells along a pseudotemporal continuum based on transcriptional similarity, we delineated transitional states and lineage commitment patterns. Integrated trajectory analysis revealed that common myeloid progenitors (CMPs) bifurcated into monocytes and macrophages during development ([Fig.5](#fig5)).

<a id="fig5"></a>
![fig5](images/5.JPG)

**Fig. 5. Pseudotime analysis for macrophage in PDAC PBMC, healthy PBMC, PDAC tissue, and adjacent normal tissuePBMCs.** 
**(A)**  Cellular trajectories in pseudotime analysis for PDAC PBMC                                                     
**(B)** Pseudotime progression for PDAC PBMC
**(C)** Cellular trajectories in  pseudotime analysis for healthy PBMC 
**(D)** Pseudotime progression for healthy PBMC

Figures a, c present the cellular trajectories in pseudotime analysis for PDAC PBMC and healthy PBMC , respectively, with cell types distinguished by different colors. Alternatively, figures b, d focus on the pseudotime progression within these same samples, where different colors now represent different stages of pseudotime.




**4.** UMAP visualization mapped spatial heterogeneity in macrophage gene expression across sample types. Transcript abundance was encoded by a continuous color gradient, with darker hues indicating elevated expression. In PDAC-associated macrophages, GZMA, COL1A1, and PF4 exhibited broad expression, while NKG7, VCAN, CCL5, CPA1, CPB1, and CTRB1 showed restricted ranges ([Fig.6,A](#figure6)). Healthy macrophages predominantly expressed GZMK, LCN2, GZMB, FGFBP2, and PRF1 (Figure 6a). PDAC tissues were enriched for S100A9, APOE, and CCL3, whereas adjacent normal regions overexpressed STC1, MALAT1, and SPP1 ([Fig.7,A And B](#fig7)).

<a id="fig7"></a>
![fig7](images/7.JPG)

**Fig. 6. Pseudotime analysis for macrophage in PDAC PBMC, healthy PBMC, PDAC tissue, and adjacent normal tissuePBMCs.** 
**(A)**  Cellular trajectories in pseudotime analysis for PDAC PBMC                                                     
**(B)** Pseudotime progression for PDAC PBMC
**(C)** Cellular trajectories in  pseudotime analysis for healthy PBMC 
**(D)** Pseudotime progression for healthy PBMC

Figures a, c present the cellular trajectories in pseudotime analysis for PDAC PBMC and healthy PBMC , respectively, with cell types distinguished by different colors. Alternatively, figures b, d focus on the pseudotime progression within these same samples, where different colors now represent different stages of pseudotime.

<a id="figure7"></a>
![figure7](images/7-1.JPG)
![figure7](images/7-2.JPG)  


**Fig. 7. Pseudotime analysis for macrophage in PDAC PBMC, healthy PBMC, PDAC tissue, and adjacent normal tissuePBMCs.** 
**(A)**  Cellular trajectories in pseudotime analysis for PDAC PBMC                                                     
**(B)** Pseudotime progression for PDAC PBMC
**(C)** Cellular trajectories in  pseudotime analysis for healthy PBMC 
**(D)** Pseudotime progression for healthy PBMC

Figures a, c present the cellular trajectories in pseudotime analysis for PDAC PBMC and healthy PBMC , respectively, with cell types distinguished by different colors. Alternatively, figures b, d focus on the pseudotime progression within these same samples, where different colors now represent different stages of pseudotime.

<a id="figure8"></a>
![figure8](images/8.JPG)  


**Fig. 8. Pseudotime analysis for macrophage in PDAC PBMC, healthy PBMC, PDAC tissue, and adjacent normal tissuePBMCs.** 
**(A)**  Cellular trajectories in pseudotime analysis for PDAC PBMC                                                     
**(B)** Pseudotime progression for PDAC PBMC
**(C)** Cellular trajectories in  pseudotime analysis for healthy PBMC 
**(D)** Pseudotime progression for healthy PBMC

Figures a, c present the cellular trajectories in pseudotime analysis for PDAC PBMC and healthy PBMC , respectively, with cell types distinguished by different colors. Alternatively, figures b, d focus on the pseudotime progression within these same samples, where different colors now represent different stages of pseudotime.

<a id="figure9"></a>
![figure9](images/9.JPG)

**Fig. 9. Pseudotime analysis for macrophage in PDAC PBMC, healthy PBMC, PDAC tissue, and adjacent normal tissuePBMCs.** 
**(A)**  Cellular trajectories in pseudotime analysis for PDAC PBMC                                                     
**(B)** Pseudotime progression for PDAC PBMC
**(C)** Cellular trajectories in  pseudotime analysis for healthy PBMC 
**(D)** Pseudotime progression for healthy PBMC

Figures a, c present the cellular trajectories in pseudotime analysis for PDAC PBMC and healthy PBMC , respectively, with cell types distinguished by different colors. Alternatively, figures b, d focus on the pseudotime progression within these same samples, where different colors now represent different stages of pseudotime.



**5.** DESeq2-based differential expression analysis identified three genes significantly upregulated in PDAC PBMCs versus healthy controls (Figure 10a). In PDAC tissues, 18 genes were upregulated alongside broad downregulation of multiple loci (Figure 10b). Gene Ontology (GO) enrichment revealed distinct pathway activation:
- Tissues: Ribosomal biogenesis (small subunit), rRNA processing, cytoplasmic translation, and amino acid metabolism were robustly activated. Conversely, pathways regulating multicellular processes, signal transduction, and stress responses were suppressed ([Fig.11,B And D](#fig)).
- PBMCs: Cytoplasmic translation and protein metabolism dominated activated pathways, while macromolecule biosynthesis showed partial suppression ([Fig.11,A and C](#fig11)).

<a id="figure10"></a>
![figure10](images/10.JPG)

**Fig. 10. Pseudotime analysis for macrophage in PDAC PBMC, healthy PBMC, PDAC tissue, and adjacent normal tissuePBMCs.** 
**(A)**  Cellular trajectories in pseudotime analysis for PDAC PBMC                                                     
**(B)** Pseudotime progression for PDAC PBMC
**(C)** Cellular trajectories in  pseudotime analysis for healthy PBMC 
**(D)** Pseudotime progression for healthy PBMC


<a id="fig11"></a>
![fig11](images/11.JPG)

**Fig. 11. Pseudotime analysis for macrophage in PDAC PBMC, healthy PBMC, PDAC tissue, and adjacent normal tissuePBMCs.** 
**(A)**  Cellular trajectories in pseudotime analysis for PDAC PBMC                                                     
**(B)** Pseudotime progression for PDAC PBMC
**(C)** Cellular trajectories in  pseudotime analysis for healthy PBMC 
**(D)** Pseudotime progression for healthy PBMC




**6.** Spatial transcriptomic profiling resolved the topographic relationship between macrophages and tumor cells, demonstrating uniform tissue distribution (Figure 123a). High-resolution analysis identified NOP53 as the most abundantly expressed gene, followed by CFB, PPA1, CFTR, GATM, SERPINB1, and ANPEP (Figures 123b, 13a).

<a id="fig12"></a>
![fig11](images/12.JPG)

**Fig. 12. Pseudotime analysis for macrophage in PDAC PBMC, healthy PBMC, PDAC tissue, and adjacent normal tissuePBMCs.** 
**(A)**  Cellular trajectories in pseudotime analysis for PDAC PBMC                                                     
**(B)** Pseudotime progression for PDAC PBMC
**(C)** Cellular trajectories in  pseudotime analysis for healthy PBMC 
**(D)** Pseudotime progression for healthy PBMC



#### Key Innovations
- Multi-Omics Integration: Harmonized single-cell and spatial transcriptomics to dissect cell type-specific dynamics.
- Trajectory Inference: Uncovered myeloid lineage commitment via pseudotemporal ordering.
- Pathway Prioritization: Identified ribosomal biogenesis and stress response as central regulatory nodes in PDAC.
This systematic framework advances precision oncology by decoding transcriptional circuits driving PDAC progression and microenvironment remodeling.



#### Discussion
In this study, it was found that the gene expression differences between PDAC PBMCs and healthy PBMCs were not significant, but there were notable differences in gene expression between PDAC tissues and adjacent normal tissues. The differential gene expression analysis reveals that, although there are relatively small differences in gene expression between DAC PBMCs (Peripheral Blood Mononuclear Cells from patients with ductal adenocarcinoma of the pancreas) and healthy PBMCs, the results of gene enrichment analysis indicate that genes related to cytoplasmic translation, translation, protein metabolic process, gene expression, and macromolecule biosynthetic process are activated. Indeed, PDAC does cause alterations in gene expression within PBMCs, but these changes are not significant. The gene expression profiles of PDAC (pancreatic ductal adenocarcinoma) tissues differ significantly from those of adjacent healthy tissues. 

Spatial transcriptomics results reveal an even distribution of macrophages and tumor cells within the tissues. Notably, several genes, including NOP53, CFB, PPA1, CFTR, GATM, SERPINB1, and ANPEP, are highly expressed in these tissues, with NOP53 exhibiting the highest expression level. Specifically, NOP53 is abundantly expressed in PDAC tissues and functions as a tumor suppressor gene. Following DNA damage in humans, the interaction between NOP53 and RPL11 is disrupted, causing RPL11 to translocate to the nucleoplasm. This translocation subsequently inhibits MDM2 and activates p53, thereby suppressing cell division. Additionally, NOP53 expression impedes the PI3K-AKT signaling pathway, promoting endogenous apoptosis and inhibiting anabolic metabolism in cells. Thus, NOP53's abundant expression in PDAC tissues serves to suppress the division and proliferation of cancer cells. SDPR gene is also expressed at higher levels in PDAC tissues compared to adjacent healthy tissues, acting as another tumor suppressor gene. It participates in cellular signal transduction by blocking the TGF-β signaling pathway, which inhibits the production of Treg cells and subsequently suppresses the epithelial-mesenchymal transition (EMT) phenotype in breast cancer cells, although its direct role in PDAC may require further investigation. STC1 is widely expressed in macrophages within tumor tissues and can inhibit their phagocytosis of dead tumor cells, thereby suppressing their antigen-presenting capacity and inhibiting the activation of T and B cells. In gene differential expression analyses, elevated levels of S100A4 were observed in both PBMCs from PDAC patients and in PDAC tissues. S100A4 can inhibit the translocation of p53 protein from the cytoplasm to the nucleus, reducing nuclear p53 protein levels, and also inhibit the transcriptional activity of p53, thereby promoting PDAC progression. STC1 can upregulate S100A4 expression by promoting the phosphorylation of EGFR and ERK signals, exacerbating the malignancy of PDAC. The expression of SPP1 is upregulated in PDAC tissues. The CXCL9:SPP1 ratio plays a role in determining the polarity of tumor cells, and the upregulation of SPP1 expression may promote tumor growth, invasion, and metastasis. However, the expression product of SPP1 is also a cytokine that can upregulate the expression of interferon-γ (IFN-γ) and interleukin-12 (IL-12) while reducing the production of interleukin-10 (IL-10). IFN-γ and IL-12 are important antitumor immune factors that can activate immune cells and enhance their killing effect on tumor cells. In contrast, IL-10 is an anti-inflammatory factor that can inhibit the activity of immune cells, thereby favoring tumor growth and evading immune surveillance. Thus, SPP1 can both promote and inhibit the development of PDAC.

In PDAC tissues, changes in gene expression lead to a reduction in the cytotoxicity of macrophages. By examining the differences in gene expression between tissue samples and healthy tissues, if the expression levels of genes such as NOP53, STC1, and SPP1 are elevated, it may indicate the presence of PDAC. This can be used for early diagnosis of cancer occurrence, facilitating subsequent treatment.

#### Methods
##### Data Acquisition
The single-cell data from human samples were sourced from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE155698. The dataset was derived from high-throughput sequencing-based expression profiling analysis. The repository encompasses 16 PDAC tissue specimens, 3 adjacent normal pancreatic tissue samples, 16 PBMC samples obtained from human PDAC patients, and 4 PBMC samples from healthy volunteers. The tissues were subjected to mechanical mincing and enzymatic digestion using collagenase P (1 mg/mL DMEM), followed by filtration through a 40 μm mesh to isolate single cells. Subsequently, dead cells were eliminated utilizing the MACS® Dead Cell Removal Kit (Miltenyi Biotec Inc.). Single-cell cDNA libraries were constructed and sequenced on the 10x Genomics platform at the University of Michigan Sequencing Core. The sequencing was performed on HiSeq 4000 or NovaSeq 6000 (Illumina) instruments, employing paired-end 50-cycle reads with a sequencing depth of 100,000 reads per sample. The raw sequencing data were processed and aligned by the University of Michigan DNA Sequencing Core. Cellranger count version 3.0.0 was employed with default parameters, with an initial anticipated cell count set at 10,000.

The spatial transcriptomic dataset for human pancreatic cancer samples was obtained from the 10x Genomics repository (https://www.10xgenomics.com/datasets/pancreatic-cancer-with-xenium-human-multi-tissue-and-cancer-panel-1-standard). Formalin-Fixed Paraffin-Embedded (FFPE) tissue blocks, characterized by adenocarcinoma of grade I-II with approximately 50% tumor content, were sourced from AcePix Biosciences. Tissue preparation was conducted in accordance with the demonstrated protocols outlined in the Xenium In Situ Tissue Preparation Guide for FFPE (CG000578) and Xenium In Situ Dewaxing and Decrosslinking for FFPE Tissue (CG000580). The gene panel utilized was the Xenium Human Multi-Tissue and Cancer Panel, which was pre-designed by 10x Genomics and informed by single-cell RNA sequencing data that were curated and standardized by the Human Protein Atlas. The panel comprises 93 genes specifically selected to detect major cell types in breast, lung, kidney, liver, skin, colon, heart, lymph node, and pancreas tissues, as well as 284 genes designed to be common across all tissues, with an emphasis on immune and cancer markers. This pre-designed panel was further supplemented with an additional 97 genes, including high expressers, to evaluate potential sensitivity loss due to gene crowding. The Xenium Analyzer was operated following the Xenium Analyzer User Guide (CG000584), and the on-instrument analysis was conducted using Xenium Onboard Analysis version 1.6.0.
##### Filtering and normalization of scRNA-seq data：
In this experiment, Seurat v5 (https://satijalab.org/seurat/) was utilized. Cells with less than or more than 200 expressed genes and genes expressed in fewer than 3 cells were filtered out. Cells with mitochondrial gene expression exceeding 5% were also filtered out.The remaining cells and gene subsets were used for subsequent analysis.
##### Dimensional reduction and clustering：
Integrated analysis was conducted using Harmony for four healthy PBMC(Peripheral Blood Mononuclear Cells) samples, four PBMC samples from patients with PDAC(pancreatic ductal adenocarcinoma), four PDAC tissue samples, and three adjacent normal tissue samples. Additionally, cell annotation was performed for each sample using SingleR. During the clustering process with Harmony integration, the top 30 dimensions were selected. The dimensionality reduction method used was Principal Component Analysis (PCA).
##### Pseudotime analysis: 
The subsets of cell types including macrophages, monocytes, and CMPs were extracted from each of the integrated files using the subset method, resulting in integrated data for these three cell types across the four integrated datasets. Then, CMP was selected as the root cell, and Monocle3 was used to learn the cellular trajectory. After learning the cellular trajectory, the top 10 genes expressed in macrophages were selected for differential gene expression analysis.
##### Gene Expression Differential Analysis: 
The logFC values for each gene in a total of 41 samples were obtained using FindAllMarkers, with the parameter threshold set to 0.25. The samples were divided into two groups: adjacent normal  tissues versus PDAC, and healthy PBMC versus PDAC PBMC, for differential gene expression analysis.
##### Gene Enrichment Analysis: 
Gene enrichment analysis was conducted on the data obtained from the differential gene expression analysis using both GO and KEGG excluding data with a p-value less than 0.05, respectively. 


### References
<a id="ref1"></a>[1] Chouari, T. et al. Advances in Immunotherapeutics in Pancreatic Ductal Adenocarcinoma. Cancers 15, 4265 (2023).
<a id="ref2"></a>[2] Ilic, M. & Ilic, I. Epidemiology of pancreatic cancer. World J. Gastroenterol. 22, 9694 (2016).
<a id="ref3"></a>[3] Klein, A. P. Pancreatic cancer epidemiology: understanding the role of lifestyle and inherited risk factors. Nat. Rev. Gastroenterol. Hepatol. 18, 493–502 (2021).
<a id="ref4"></a>[4] Ho, T. T. B. et al. Combination of gemcitabine and anti-PD-1 antibody enhances the anticancer effect of M1 macrophages and the Th1 response in a murine model of pancreatic cancer liver metastasis. J. Immunother. Cancer 8, e001367 (2020).
<a id="ref5"></a>[5] Zhu, Y. et al. CSF1/CSF1R Blockade Reprograms Tumor-Infiltrating Macrophages and Improves Response to T-cell Checkpoint Immunotherapy in Pancreatic Cancer Models. Cancer Res. 74, 5057–5069 (2014).
<a id="ref6"></a>[6] Locati, M., Curtale, G. & Mantovani, A. Diversity, Mechanisms, and Significance of Macrophage Plasticity. Annu. Rev. Pathol. Mech. Dis. 15, 123–147 (2020).
<a id="ref7"></a>[7] Mantovani, A., Marchesi, F., Malesci, A., Laghi, L. & Allavena, P. Tumour-associated macrophages as treatment targets in oncology. Nat. Rev. Clin. Oncol. 14, 399–416 (2017).
<a id="ref8"></a>[8] Wu, Y. et al. Macrophage cell membrane-based nanoparticles: a new promising biomimetic platform for targeted delivery and treatment. J. Nanobiotechnology 20, 542 (2022).
<a id="ref9"></a>[9] Wynn, T. A., Chawla, A. & Pollard, J. W. Macrophage biology in development, homeostasis and disease. Nature 496, 445–455 (2013).
<a id="ref10"></a>[10] Cai, H., Zhang, Y., Wang, J. & Gu, J. Defects in Macrophage Reprogramming in Cancer Therapy: The Negative Impact of PD-L1/PD-1. Front. Immunol. 12, 690869 (2021).
<a id="ref11"></a>[11] Ruffell, B. & Coussens, L. M. Macrophages and Therapeutic Resistance in Cancer. Cancer Cell 27, 462–472 (2015).
<a id="ref12"></a>[12] DeNardo, D. G. & Ruffell, B. Macrophages as regulators of tumour immunity and immunotherapy. Nat. Rev. Immunol. 19, 369–382 (2019).
<a id="ref13"></a>[13] Ngambenjawong, C., Gustafson, H. H. & Pun, S. H. Progress in tumor-associated macrophage (TAM)-targeted therapeutics. Adv. Drug Deliv. Rev. 114, 206–221 (2017).
<a id="ref14"></a>[14] Dai, E. et al. Autophagy-dependent ferroptosis drives tumor-associated macrophage polarization via release and uptake of oncogenic KRAS protein. Autophagy 16, 2069–2083 (2020).
<a id="ref15"></a>[15] Mehla, K. & Singh, P. K. Metabolic Regulation of Macrophage Polarization in Cancer. Trends Cancer 5, 822–834 (2019).
<a id="ref16"></a>[16] Fendl, B., Berghoff, A. S., Preusser, M. & Maier, B. Macrophage and monocyte subsets as new therapeutic targets in cancer immunotherapy. ESMO Open 8, 100776 (2023).
<a id="ref17"></a>[17] Wei, C. et al. Crosstalk between cancer cells and tumor associated macrophages is required for mesenchymal circulating tumor cell-mediated colorectal cancer metastasis. Mol. Cancer 18, 64 (2019).
<a id="ref18"></a>[18] Wu, K. et al. Redefining Tumor-Associated Macrophage Subpopulations and Functions in the Tumor Microenvironment. Front. Immunol. 11, 1731 (2020).
<a id="ref19"></a>[19] Seyfried, T. N. & Huysentruyt, L. C. On the Origin of Cancer Metastasis. Crit. Rev. Oncog. 18, 43–73 (2013).
<a id="ref20"></a>[20] Cao, L., Meng, X., Zhang, Z., Liu, Z. & He, Y. Macrophage heterogeneity and its interactions with stromal cells in tumour microenvironment. Cell Biosci. 14, 16 (2024).
<a id="ref21"></a>[21] Zhang, X., Ji, L. & Li, M. O. Control of tumor-associated macrophage responses by nutrient acquisition and metabolism. Immunity 56, 14–31 (2023).
<a id="ref22"></a>[22] Rao, L. et al. Activating Macrophage‐Mediated Cancer Immunotherapy by Genetically Edited Nanoparticles. Adv. Mater. 32, 2004853 (2020).
<a id="ref23"></a>[23] Gao, J., Liang, Y. & Wang, L. Shaping Polarization Of Tumor-Associated Macrophages In Cancer Immunotherapy. Front. Immunol. 13, 888713 (2022).
<a id="ref24"></a>[24] Mills, C. D., Lenz, L. L. & Harris, R. A. A Breakthrough: Macrophage-Directed Cancer Immunotherapy. Cancer Res. 76, 513–516 (2016).
<a id="ref25"></a>[25] Wang, H., Yung, M. M. H., Ngan, H. Y. S., Chan, K. K. L. & Chan, D. W. The Impact of the Tumor Microenvironment on Macrophage Polarization in Cancer Metastatic Progression. Int. J. Mol. Sci. 22, 6560 (2021).
<a id="ref26"></a>[26] Li, M. et al. Metabolism, metabolites, and macrophages in cancer. J. Hematol. Oncol.J Hematol Oncol 16, 80 (2023).
<a id="ref27"></a>[27] Halbrook, C. J. et al. Macrophage-Released Pyrimidines Inhibit Gemcitabine Therapy in Pancreatic Cancer. Cell Metab. 29, 1390-1399.e6 (2019).
<a id="ref28"></a>[28] Menjivar, R. E. et al. Arginase 1 is a key driver of immune suppression in pancreatic cancer. eLife 12, e80721 (2023).
<a id="ref29"></a>[29] Kuen, J., Darowski, D., Kluge, T. & Majety, M. Pancreatic cancer cell/fibroblast co-culture induces M2 like macrophages that influence therapeutic response in a 3D model. PLOS ONE 12, e0182039 (2017).
<a id="ref30"></a>[30] LaRue, M. M. et al. Metabolic reprogramming of tumor-associated macrophages by collagen turnover promotes fibrosis in pancreatic cancer. Proc. Natl. Acad. Sci. 119, e2119168119 (2022).
<a id="ref31"></a>[31] Kemp, S. B. et al. Apolipoprotein E Promotes Immune Suppression in Pancreatic Cancer through NF-κB–Mediated Production of CXCL1. Cancer Res. 81, 4305–4318 (2021).
<a id="ref32"></a>[32] Kemp, S. B. et al. Pancreatic cancer is marked by complement-high blood monocytes and tumor-associated macrophages. Life Sci. Alliance 4, e202000935 (2021).
<a id="ref33"></a>[33] Lee, B. Y. et al. Heterocellular OSM-OSMR signalling reprograms fibroblasts to promote pancreatic cancer growth and metastasis. Nat. Commun. 12, 7336 (2021).
<a id="ref34"></a>[34] Padoan, A., Plebani, M. & Basso, D. Inflammation and Pancreatic Cancer: Focus on Metabolism, Cytokines, and Immunity. Int. J. Mol. Sci. 20, 676 (2019).
<a id="ref35"></a>[35] Lankadasari, M. B., Mukhopadhyay, P., Mohammed, S. & Harikumar, K. B. TAMing pancreatic cancer: combat with a double edged sword. Mol. Cancer 18, 48 (2019).
<a id="ref36"></a>[36] Yang, Y. et al. M2 Macrophage-Derived Exosomes Promote Angiogenesis and Growth of Pancreatic Ductal Adenocarcinoma by Targeting E2F2. Mol. Ther. 29, 1226–1238 (2021).
<a id="ref37"></a>[37] Gross, A. et al. Technologies for Single-Cell Isolation. Int. J. Mol. Sci. 16, 16897–16919 (2015).
<a id="ref38"></a>[38] Nasir, I. et al. Tumor macrophage functional heterogeneity can inform the development of novel cancer therapies. Trends Immunol. 44, 971–985 (2023).
<a id="ref39"></a>[39] Yang, K., Yang, T., Yu, J., Li, F. & Zhao, X. Integrated transcriptional analysis reveals macrophage heterogeneity and macrophage-tumor cell interactions in the progression of pancreatic ductal adenocarcinoma. BMC Cancer 23, 199 (2023).
<a id="ref40"></a>[40] Marx, V. Method of the Year: spatially resolved transcriptomics. Nat. Methods 18, 9–14 (2021).
<a id="ref41"></a>[41] Gerlinger, M. et al. Intratumor Heterogeneity and Branched Evolution Revealed by Multiregion Sequencing. N. Engl. J. Med. 366, 883–892 (2012).