<!doctype html>
<html lang="en-US">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <meta name="generator" content="VuePress 2.0.0-rc.20" />
    <style>
      :root {
        --vp-c-bg: #fff;
      }

      [data-theme='dark'] {
        --vp-c-bg: #1b1b1f;
      }

      html,
      body {
        background-color: var(--vp-c-bg);
      }
    </style>
    <script>
      const useChoice = localStorage.getItem('vuepress-color-scheme')
      const systemStatus =
        'matchMedia' in window
          ? window.matchMedia('(prefers-color-scheme: dark)').matches
          : false

      if (useChoice === 'light') {
        document.documentElement.dataset.theme = 'light'
      } else if (useChoice === 'dark' || systemStatus) {
        document.documentElement.dataset.theme = 'dark'
      }
    </script>
    <title>This Set of Bioinformatics Analysis Code from Harvard University's Nature Paper is Worth Learning | Mengfei Li</title><meta name="description" content="Recently, a research team from Harvard University published an important study in the journal Nature. This research delved into the developmental trajectory of neurons in the mammalian hypothalamic preoptic area (POA), revealing the significant influence of factors such as sensory input, sex, and function on their development. It provides a new perspective for understanding the neural basis of instinctive behaviors.">
    <link rel="preload" href="/assets/style-CMC_ymkA.css" as="style"><link rel="stylesheet" href="/assets/style-CMC_ymkA.css">
    <link rel="modulepreload" href="/assets/app-CrqbF7DL.js"><link rel="modulepreload" href="/assets/article-2.html-BPJZKmPw.js">
    <link rel="prefetch" href="/assets/cv.html-D4hZpk7A.js" as="script"><link rel="prefetch" href="/assets/index.html-ClVTB5uf.js" as="script"><link rel="prefetch" href="/assets/article-1.html-Tfi4G0Jc.js" as="script"><link rel="prefetch" href="/assets/1.html-DI4ACmmg.js" as="script"><link rel="prefetch" href="/assets/article-1.html-D5RBjVwf.js" as="script"><link rel="prefetch" href="/assets/article-2.html-BdbedaX9.js" as="script"><link rel="prefetch" href="/assets/article-3.html-BomlOZjU.js" as="script"><link rel="prefetch" href="/assets/article-4.html-D1MxE_nk.js" as="script"><link rel="prefetch" href="/assets/article-5.html-CdsWaU_6.js" as="script"><link rel="prefetch" href="/assets/article-6.html-D44nD4iF.js" as="script"><link rel="prefetch" href="/assets/cv.html-CTyLLbPV.js" as="script"><link rel="prefetch" href="/assets/index.html-F7GqOdIs.js" as="script"><link rel="prefetch" href="/assets/plan.html-CQXXKQl8.js" as="script"><link rel="prefetch" href="/assets/article-1.html-DnuBTomC.js" as="script"><link rel="prefetch" href="/assets/article-2.html-8BlJOA6u.js" as="script"><link rel="prefetch" href="/assets/article-3.html-Dc5D2MVI.js" as="script"><link rel="prefetch" href="/assets/article-1.html-poUP5kYg.js" as="script"><link rel="prefetch" href="/assets/article-2.html-8hP4YJkL.js" as="script"><link rel="prefetch" href="/assets/article-1.html-IwfSEP-D.js" as="script"><link rel="prefetch" href="/assets/article-2.html-MJBuBA1D.js" as="script"><link rel="prefetch" href="/assets/article-3.html-B99m_gfp.js" as="script"><link rel="prefetch" href="/assets/1.html-0u3mxAwn.js" as="script"><link rel="prefetch" href="/assets/article-1.html-D5qhQQ2I.js" as="script"><link rel="prefetch" href="/assets/article-2.html-DDEeC9Zr.js" as="script"><link rel="prefetch" href="/assets/article-3.html-B8I1t9GD.js" as="script"><link rel="prefetch" href="/assets/article-4.html--i3MTMqY.js" as="script"><link rel="prefetch" href="/assets/article-5.html-BwUMKwFG.js" as="script"><link rel="prefetch" href="/assets/article-6.html-DhpK7ydG.js" as="script"><link rel="prefetch" href="/assets/404.html-CloqwltV.js" as="script"><link rel="prefetch" href="/assets/index.html-DLhWVlki.js" as="script"><link rel="prefetch" href="/assets/index.html-DdEZbNfG.js" as="script"><link rel="prefetch" href="/assets/index.html-LSRtop2d.js" as="script"><link rel="prefetch" href="/assets/index.html-tjAH3zVe.js" as="script"><link rel="prefetch" href="/assets/index.html-Cj47_jwY.js" as="script"><link rel="prefetch" href="/assets/index.html-HzFyn0W1.js" as="script"><link rel="prefetch" href="/assets/setupDevtools-QXEOFQJV-Bnm_Rcsg.js" as="script">
  </head>
  <body>
    <div id="app"><!--[--><div class="vp-theme-container external-link-icon" vp-container data-v-a3194778><!--[--><header class="vp-navbar" vp-navbar><div class="vp-toggle-sidebar-button" title="toggle sidebar" aria-expanded="false" role="button" tabindex="0"><div class="icon" aria-hidden="true"><span></span><span></span><span></span></div></div><span><a class="route-link" href="/"><img class="vp-site-logo" src="/images/hero.png" alt="Mengfei Li"><span class="vp-site-name vp-hide-mobile" aria-hidden="true">Mengfei Li</span></a></span><div class="vp-navbar-items-wrapper" style=""><!--[--><!--]--><nav class="vp-navbar-items vp-hide-mobile" aria-label="site navigation"><!--[--><div class="vp-navbar-item"><a class="route-link auto-link" href="/" aria-label="Home"><!--[--><!--[--><!--]--><!--]-->Home<!--[--><!--[--><!--]--><!--]--></a></div><div class="vp-navbar-item"><a class="route-link auto-link" href="/researchList/" aria-label="Research"><!--[--><!--[--><!--]--><!--]-->Research<!--[--><!--[--><!--]--><!--]--></a></div><div class="vp-navbar-item"><a class="route-link auto-link" href="/repositoryList/" aria-label="Repositories"><!--[--><!--[--><!--]--><!--]-->Repositories<!--[--><!--[--><!--]--><!--]--></a></div><div class="vp-navbar-item"><a class="route-link auto-link" href="/blogList/" aria-label="Blog"><!--[--><!--[--><!--]--><!--]-->Blog<!--[--><!--[--><!--]--><!--]--></a></div><div class="vp-navbar-item"><a class="route-link auto-link" href="/cv.html" aria-label="CV"><!--[--><!--[--><!--]--><!--]-->CV<!--[--><!--[--><!--]--><!--]--></a></div><div class="vp-navbar-item"><div class="vp-navbar-dropdown-wrapper"><button class="vp-navbar-dropdown-title" type="button" aria-label="Language"><span class="title">Language</span><span class="arrow down"></span></button><button class="vp-navbar-dropdown-title-mobile" type="button" aria-label="Language"><span class="title">Language</span><span class="right arrow"></span></button><ul style="display:none;" class="vp-navbar-dropdown"><!--[--><li class="vp-navbar-dropdown-item"><a class="route-link route-link-active auto-link" href="/blog/article-2.html" aria-label="English"><!--[--><!--[--><!--]--><!--]-->English<!--[--><!--[--><!--]--><!--]--></a></li><li class="vp-navbar-dropdown-item"><a class="route-link auto-link" href="/zh/blog/article-2.html" aria-label="中文"><!--[--><!--[--><!--]--><!--]-->中文<!--[--><!--[--><!--]--><!--]--></a></li><!--]--></ul></div></div><!--]--></nav><!--[--><!--]--><button type="button" class="vp-toggle-color-mode-button" title="toggle color mode"><svg class="light-icon" viewbox="0 0 32 32" style=""><path d="M16 12.005a4 4 0 1 1-4 4a4.005 4.005 0 0 1 4-4m0-2a6 6 0 1 0 6 6a6 6 0 0 0-6-6z" fill="currentColor"></path><path d="M5.394 6.813l1.414-1.415l3.506 3.506L8.9 10.318z" fill="currentColor"></path><path d="M2 15.005h5v2H2z" fill="currentColor"></path><path d="M5.394 25.197L8.9 21.691l1.414 1.415l-3.506 3.505z" fill="currentColor"></path><path d="M15 25.005h2v5h-2z" fill="currentColor"></path><path d="M21.687 23.106l1.414-1.415l3.506 3.506l-1.414 1.414z" fill="currentColor"></path><path d="M25 15.005h5v2h-5z" fill="currentColor"></path><path d="M21.687 8.904l3.506-3.506l1.414 1.415l-3.506 3.505z" fill="currentColor"></path><path d="M15 2.005h2v5h-2z" fill="currentColor"></path></svg><svg class="dark-icon" viewbox="0 0 32 32" style="display:none;"><path d="M13.502 5.414a15.075 15.075 0 0 0 11.594 18.194a11.113 11.113 0 0 1-7.975 3.39c-.138 0-.278.005-.418 0a11.094 11.094 0 0 1-3.2-21.584M14.98 3a1.002 1.002 0 0 0-.175.016a13.096 13.096 0 0 0 1.825 25.981c.164.006.328 0 .49 0a13.072 13.072 0 0 0 10.703-5.555a1.01 1.01 0 0 0-.783-1.565A13.08 13.08 0 0 1 15.89 4.38A1.015 1.015 0 0 0 14.98 3z" fill="currentColor"></path></svg></button><!----></div></header><!--]--><div class="vp-sidebar-mask"></div><!--[--><aside class="vp-sidebar" vp-sidebar><nav class="vp-navbar-items" aria-label="site navigation"><!--[--><div class="vp-navbar-item"><a class="route-link auto-link" href="/" aria-label="Home"><!--[--><!--[--><!--]--><!--]-->Home<!--[--><!--[--><!--]--><!--]--></a></div><div class="vp-navbar-item"><a class="route-link auto-link" href="/researchList/" aria-label="Research"><!--[--><!--[--><!--]--><!--]-->Research<!--[--><!--[--><!--]--><!--]--></a></div><div class="vp-navbar-item"><a class="route-link auto-link" href="/repositoryList/" aria-label="Repositories"><!--[--><!--[--><!--]--><!--]-->Repositories<!--[--><!--[--><!--]--><!--]--></a></div><div class="vp-navbar-item"><a class="route-link auto-link" href="/blogList/" aria-label="Blog"><!--[--><!--[--><!--]--><!--]-->Blog<!--[--><!--[--><!--]--><!--]--></a></div><div class="vp-navbar-item"><a class="route-link auto-link" href="/cv.html" aria-label="CV"><!--[--><!--[--><!--]--><!--]-->CV<!--[--><!--[--><!--]--><!--]--></a></div><div class="vp-navbar-item"><div class="vp-navbar-dropdown-wrapper"><button class="vp-navbar-dropdown-title" type="button" aria-label="Language"><span class="title">Language</span><span class="arrow down"></span></button><button class="vp-navbar-dropdown-title-mobile" type="button" aria-label="Language"><span class="title">Language</span><span class="right arrow"></span></button><ul style="display:none;" class="vp-navbar-dropdown"><!--[--><li class="vp-navbar-dropdown-item"><a class="route-link route-link-active auto-link" href="/blog/article-2.html" aria-label="English"><!--[--><!--[--><!--]--><!--]-->English<!--[--><!--[--><!--]--><!--]--></a></li><li class="vp-navbar-dropdown-item"><a class="route-link auto-link" href="/zh/blog/article-2.html" aria-label="中文"><!--[--><!--[--><!--]--><!--]-->中文<!--[--><!--[--><!--]--><!--]--></a></li><!--]--></ul></div></div><!--]--></nav><!--[--><!--]--><ul class="vp-sidebar-items"><!--[--><li><p tabindex="0" class="vp-sidebar-item vp-sidebar-heading">This Set of Bioinformatics Analysis Code from Harvard University&#39;s Nature Paper is Worth Learning <!----></p><!----></li><!--]--></ul><!--[--><!--]--></aside><!--]--><!--[--><div class="vp-page blog-page-wrapper" data-v-a3194778><aside class="vp-blog-info-wrapper" style="transition:transform 0.25s ease-in-out 0.16s, opacity 0.25s ease-in-out 0.16s;transform:translateY(0px);opacity:1;" data-v-a3194778><div class="vp-blogger-info" vocab="https://schema.org/" typeof="Person"><div class="vp-blogger" data-balloon-pos="down"><img class="vp-blogger-avatar" src="/images/profile.png" property="image" alt="Blogger Avatar" loading="lazy"><div class="vp-blogger-name" property="name">Mengfei Li</div><div class="vp-blogger-description">biological science</div><div class="vp-blogger-description">Undergraduate</div></div><div class="vp-social-medias"><div class="vp-social-media-box"><a class="vp-social-media-a" href="/files/cv_Mengfei_Li.pdf" rel="noopener noreferrer" target="_blank" aria-label="Gitlab" data-balloon-pos="up"><svg t="1741912630145" class="icon" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg" p-id="3990" width="32" height="32"><path d="M512 512m-512 0a512 512 0 1 0 1024 0 512 512 0 1 0-1024 0Z" fill="#D0DDFE" p-id="3991"></path><path d="M224 176m64 0l448 0q64 0 64 64l0 544q0 64-64 64l-448 0q-64 0-64-64l0-544q0-64 64-64Z" fill="#FFFFFF" p-id="3992"></path><path d="M263.792 225.328m64 0l368.4 0q64 0 64 64l0 451.136q0 64-64 64l-368.4 0q-64 0-64-64l0-451.136q0-64 64-64Z" fill="#98BBFF" p-id="3993"></path><path d="M404.704 306.208a108.032 104.352 0 1 0 216.064 0 108.032 104.352 0 1 0-216.064 0Z" fill="#FFFFFF" p-id="3994"></path><path d="M489.904 343.552h45.632A164.128 164.128 0 0 1 699.68 507.68a10.256 10.256 0 0 1-10.256 10.256H336.032a10.256 10.256 0 0 1-10.256-10.256 164.128 164.128 0 0 1 164.128-164.128z" fill="#FFFFFF" p-id="3995"></path><path d="M625.632 496H398.352a25.04 25.04 0 0 1-25.008-25.008 99.936 99.936 0 0 1 75.312-97.04 92.08 92.08 0 0 0 63.152 24.992 92.096 92.096 0 0 0 63.136-24.992 99.888 99.888 0 0 1 75.616 97.04A25.04 25.04 0 0 1 625.632 496z" fill="#C9D9FF" p-id="3996"></path><path d="M512 302.512m-73.952 0a73.952 73.952 0 1 0 147.904 0 73.952 73.952 0 1 0-147.904 0Z" fill="#FFB444" p-id="3997"></path><path d="M362.576 568m27.2 0l235.168 0q27.2 0 27.2 27.2l0 0q0 27.2-27.2 27.2l-235.168 0q-27.2 0-27.2-27.2l0 0q0-27.2 27.2-27.2Z" fill="#FFFFFF" p-id="3998"></path><path d="M316.016 667.776m27.2 0l152.432 0q27.2 0 27.2 27.2l0 0q0 27.2-27.2 27.2l-152.432 0q-27.2 0-27.2-27.2l0 0q0-27.2 27.2-27.2Z" fill="#FFFFFF" p-id="3999"></path><path d="M612.349533 634.200755m20.138253 24.868671l56.820079 70.166954q20.138253 24.868671-4.730418 45.006924l0 0q-24.868671 20.138253-45.006923-4.730419l-56.82008-70.166954q-20.138253-24.868671 4.730419-45.006923l0 0q24.868671-20.138253 45.006923 4.730418Z" fill="#216CFF" p-id="4000"></path><path d="M806.073252 616.026867m-19.258081 25.556337l-95.366016 126.554977q-19.258081 25.556336-44.814417 6.298256l0 0q-25.556336-19.258081-6.298255-44.814417l95.366016-126.554978q19.258081-25.556336 44.814417-6.298255l0 0q25.556336 19.258081 6.298255 44.814417Z" fill="#216CFF" p-id="4001"></path></svg><span class="vp-social-media-text">CV Dowonload</span></a><a class="vp-social-media-a" href="mailto:202223142018@mail.sdu.edu.cn" rel="noopener noreferrer" target="_blank" aria-label="Email" data-balloon-pos="up"><svg xmlns="http://www.w3.org/2000/svg" class="vp-social-media-icon email-icon" viewBox="0 0 1024 1024"><circle cx="512" cy="512" r="512" fill="#1384FF"></circle><path fill="#fff" d="M270.077 286.233H751.99c32.933 0 59.86 24.855 60.274 55.51l-301.023 157L210.217 341.88c.207-30.723 26.927-55.717 59.86-55.717zm-59.929 115.714-.276 277.756c0 30.931 27.134 56.2 60.205 56.2H751.99c33.14 0 60.274-25.269 60.274-56.2V401.81L518.283 551.492a15.88 15.88 0 0 1-14.43 0L210.148 401.947z"></path></svg><span class="vp-social-media-text">Email</span></a><a class="vp-social-media-a" href="https://github.com/lmf04/" rel="noopener noreferrer" target="_blank" aria-label="GitHub" data-balloon-pos="up"><svg xmlns="http://www.w3.org/2000/svg" class="vp-social-media-icon github-icon" viewBox="0 0 1024 1024"><circle cx="512" cy="512" r="512" fill="#171515"></circle><path fill="#fff" d="M509.423 146.442c-200.317 0-362.756 162.42-362.756 362.8 0 160.266 103.936 296.24 248.109 344.217 18.139 3.327 24.76-7.872 24.76-17.486 0-8.613-.313-31.427-.49-61.702-100.912 21.923-122.205-48.63-122.205-48.63-16.495-41.91-40.28-53.067-40.28-53.067-32.937-22.51 2.492-22.053 2.492-22.053 36.407 2.566 55.568 37.386 55.568 37.386 32.362 55.438 84.907 39.43 105.58 30.143 3.296-23.444 12.667-39.43 23.032-48.498-80.557-9.156-165.246-40.28-165.246-179.297 0-39.604 14.135-71.988 37.342-97.348-3.731-9.178-16.18-46.063 3.556-96.009 0 0 30.46-9.754 99.76 37.19 28.937-8.048 59.97-12.071 90.823-12.211 30.807.14 61.843 4.165 90.822 12.21 69.26-46.944 99.663-37.189 99.663-37.189 19.792 49.946 7.34 86.831 3.61 96.01 23.25 25.359 37.29 57.742 37.29 97.347 0 139.366-84.82 170.033-165.637 179.013 13.026 11.2 24.628 33.342 24.628 67.182 0 48.498-.445 87.627-.445 99.521 0 9.702 6.535 20.988 24.945 17.444 144.03-48.067 247.881-183.95 247.881-344.175 0-200.378-162.442-362.798-362.802-362.798z"></path></svg><span class="vp-social-media-text">GitHub</span></a><a class="vp-social-media-a" href="https://lmf04.github.io/" rel="noopener noreferrer" target="_blank" aria-label="Gitlab" data-balloon-pos="up"><svg xmlns="http://www.w3.org/2000/svg" class="vp-social-media-icon gitlab-icon" viewBox="0 0 1024 1024"><circle cx="512" cy="512" r="512" fill="#E8F0FF"></circle><path fill="#E24329" d="m512 808 110-340H402z"></path><path fill="#FC6D26" d="M512 808 402 468H247z"></path><path fill="#E24329" d="M247 468h155l-67-204c-4-11-19-11-22 0z"></path><path fill="#FC6D26" d="m512 808 110-340h156z"></path><path fill="#FCA326" d="m778 468 33 104c3 10-1 19-9 25L512 808z"></path><path fill="#E24329" d="M778 468H622l68-204c3-11 18-11 21 0z"></path><path fill="#FCA326" d="M512 808 222 597c-8-5-12-16-9-25l34-104z"></path></svg><span class="vp-social-media-text">Website</span></a><a class="vp-social-media-a" href="https://www.en.sdu.edu.cn/" rel="noopener noreferrer" target="_blank" aria-label="Gitlab" data-balloon-pos="up"><svg t="1741831836813" class="icon" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg" p-id="9703" width="32" height="32"><path d="M512 976.213333A464.213333 464.213333 0 1 0 512 47.786667a464.213333 464.213333 0 0 0 0 928.426666z m0 47.786667a512 512 0 1 1 0-1024 512 512 0 0 1 0 1024z" fill="#009688" p-id="9704"></path><path d="M88.746667 513.365333a424.618667 424.618667 0 1 1 849.373866 0 424.618667 424.618667 0 0 1-849.373866 0z m414.72-14.336c-120.695467 0-179.882667 77.550933-179.882667 77.550934V737.28s61.576533 60.347733 189.371733 60.347733c127.7952 0 170.3936-60.347733 170.3936-60.347733V576.580267S624.162133 499.029333 503.466667 499.029333z m2.389333 264.192c-83.626667 0-151.415467-20.48-151.415467-45.943466 0-25.258667 67.7888-45.943467 151.415467-45.943467s151.552 20.48 151.552 45.943467c-0.068267 25.326933-67.925333 46.011733-151.552 46.011733zM111.8208 464.622933L291.566933 552.277333s87.04-107.656533 207.121067-107.656533c117.282133 0 233.198933 107.656533 233.198933 107.656533l65.058134-33.041066v96.733866c-10.376533 4.232533-17.749333 15.701333-17.749334 29.4912 0 14.062933 7.645867 25.6 18.090667 29.696l-19.182933 65.1264h56.7296l-19.456-65.604266c9.352533-4.778667 15.837867-15.9744 15.837866-29.0816a32.290133 32.290133 0 0 0-15.36-28.672v-107.178667l94.685867-47.854933-416.631467-226.781867-382.1568 229.512533z" fill="#009688" p-id="9705"></path></svg><span class="vp-social-media-text">Shandong University</span></a></div></div></div></aside><div class="vp-blog-main" data-v-a3194778><div class="vp-blog-main-box" data-v-a3194778><div data-v-a3194778><h1 data-v-a3194778>This Set of Bioinformatics Analysis Code from Harvard University&#39;s Nature Paper is Worth Learning</h1><p data-v-a3194778>2025.02.12</p><p data-v-a3194778>Recently, a research team from Harvard University published an important study in the journal Nature. This research delved into the developmental trajectory of neurons in the mammalian hypothalamic preoptic area (POA), revealing the significant influence of factors such as sensory input, sex, and function on their development. It provides a new perspective for understanding the neural basis of instinctive behaviors.</p><div data-v-a3194778><p>This code block is used to set up the environment and load the R packages required for subsequent analysis.</p><h1 id="this-code-is-used-to-generate-regional-correlation-matrices-and-metrics-corresponding-to-extended-data-fig-3d-k" tabindex="-1"><a class="header-anchor" href="#this-code-is-used-to-generate-regional-correlation-matrices-and-metrics-corresponding-to-extended-data-fig-3d-k"><span>This code is used to generate regional correlation matrices and metrics, corresponding to Extended Data Fig. 3d - k</span></a></h1><h1 id="using-excitatory-neurons-as-an-example-for-analysis" tabindex="-1"><a class="header-anchor" href="#using-excitatory-neurons-as-an-example-for-analysis"><span>Using excitatory neurons as an example for analysis</span></a></h1><h1 id="load-necessary-libraries" tabindex="-1"><a class="header-anchor" href="#load-necessary-libraries"><span>Load necessary libraries</span></a></h1><p>library(Seurat) # A powerful toolkit for single-cell RNA sequencing data analysis library(ggplot2) # For creating high-quality graphics library(patchwork) # For combining multiple ggplot graphics library(magrittr) # Provides the pipe operator %&gt;% library(tidyverse) # Contains several commonly used packages for data processing and visualization library(pheatmap) # For drawing heatmaps library(palr) # Provides color palettes</p><h1 id="this-code-block-is-used-to-generate-the-correlation-matrix-between-cell-types-at-a-given-age" tabindex="-1"><a class="header-anchor" href="#this-code-block-is-used-to-generate-the-correlation-matrix-between-cell-types-at-a-given-age"><span>This code block is used to generate the correlation matrix between cell types at a given age.</span></a></h1><h1 id="this-part-generates-matrices-as-shown-in-extended-data-fig-3d-e-g-h-using-fig-e-excitatory-neurons-at-age-p65-as-an-example" tabindex="-1"><a class="header-anchor" href="#this-part-generates-matrices-as-shown-in-extended-data-fig-3d-e-g-h-using-fig-e-excitatory-neurons-at-age-p65-as-an-example"><span>This part generates matrices as shown in Extended Data Fig. 3d, e, g, h, using Fig. e (excitatory neurons at age P65) as an example.</span></a></h1><h1 id="load-the-merged-data" tabindex="-1"><a class="header-anchor" href="#load-the-merged-data"><span>Load the merged data.</span></a></h1><h1 id="the-merged-data-contains-only-excitatory-neurons-the-operation-is-the-same-if-analyzing-inhibitory-neurons-just-run-separately-for-each" tabindex="-1"><a class="header-anchor" href="#the-merged-data-contains-only-excitatory-neurons-the-operation-is-the-same-if-analyzing-inhibitory-neurons-just-run-separately-for-each"><span>The merged data contains only excitatory neurons (the operation is the same if analyzing inhibitory neurons, just run separately for each).</span></a></h1><h1 id="these-data-are-merged-across-all-ages-and-then-normalized-using-sctransform-across-the-entire-dataset" tabindex="-1"><a class="header-anchor" href="#these-data-are-merged-across-all-ages-and-then-normalized-using-sctransform-across-the-entire-dataset"><span>These data are merged across all ages and then normalized using SCTransform across the entire dataset.</span></a></h1><p>excit.merge &lt;- readRDS(&quot;&quot;)</p><h1 id="extract-data-for-age-p65-from-the-merged-data" tabindex="-1"><a class="header-anchor" href="#extract-data-for-age-p65-from-the-merged-data"><span>Extract data for age P65 from the merged data</span></a></h1><p>p65 &lt;- subset(excit.merge, subset = age == &quot;p65&quot;)</p><h1 id="sort-the-cell-type-names" tabindex="-1"><a class="header-anchor" href="#sort-the-cell-type-names"><span>Sort the cell type names</span></a></h1><p>celltypes &lt;- names(table(p65$my.cell.type)) celltypes &lt;- c(celltypes[1:19], celltypes[21:28], celltypes[20], celltypes[29:37], celltypes[39:46], celltypes[38], celltypes[47], celltypes[50:57], celltypes[48:49], celltypes[58:64])</p><h1 id="reorder-the-cell-types" tabindex="-1"><a class="header-anchor" href="#reorder-the-cell-types"><span>Reorder the cell types</span></a></h1><p>new.order &lt;- c(grep(&quot;B&quot;, celltypes), grep(&quot;H&quot;, celltypes), grep(&quot;L&quot;, celltypes), grep(&quot;C&quot;, celltypes), grep(&quot;F&quot;, celltypes), grep(&quot;M&quot;, celltypes), grep(&quot;N&quot;, celltypes), grep(&quot;A&quot;, celltypes), grep(&quot;P&quot;, celltypes), grep(&quot;T&quot;, celltypes), grep(&quot;X&quot;, celltypes)) celltypes &lt;- celltypes[new.order]</p><h1 id="set-parameters-number-of-feature-genes" tabindex="-1"><a class="header-anchor" href="#set-parameters-number-of-feature-genes"><span>Set parameters: number of feature genes</span></a></h1><p>nfeatures = 20000</p><h1 id="set-the-default-assay-to-rna" tabindex="-1"><a class="header-anchor" href="#set-the-default-assay-to-rna"><span>Set the default assay to RNA</span></a></h1><p>DefaultAssay(p65) &lt;- &quot;RNA&quot;</p><h1 id="find-variable-feature-genes" tabindex="-1"><a class="header-anchor" href="#find-variable-feature-genes"><span>Find variable feature genes</span></a></h1><p>p65 &lt;- FindVariableFeatures(p65, nfeatures = nfeatures)</p><h1 id="scale-the-data" tabindex="-1"><a class="header-anchor" href="#scale-the-data"><span>Scale the data</span></a></h1><p>p65 &lt;- ScaleData(p65)</p><h1 id="generate-the-centroid-matrix-which-will-serve-as-input-for-the-corr-function" tabindex="-1"><a class="header-anchor" href="#generate-the-centroid-matrix-which-will-serve-as-input-for-the-corr-function"><span>Generate the centroid matrix, which will serve as input for the corr() function</span></a></h1><p>centroid.mat &lt;- matrix(, nrow = length(celltypes), ncol = nfeatures) for (i in 1:length(celltypes)) {</p><h1 id="print-progress-information-every-10-cell-types" tabindex="-1"><a class="header-anchor" href="#print-progress-information-every-10-cell-types"><span>Print progress information every 10 cell types</span></a></h1><p>if (i %% 10 == 0) { print(paste0(&quot;on cell type number &quot;, i)) }</p><h1 id="skip-cell-types-with-fewer-than-2-cells" tabindex="-1"><a class="header-anchor" href="#skip-cell-types-with-fewer-than-2-cells"><span>Skip cell types with fewer than 2 cells</span></a></h1><p>if (table(p65$my.cell.type)[i] &lt; 2) { print(&quot;NOTE: this cell type has less than 2 cells. skipping...&quot;) next }</p><h1 id="extract-data-for-the-current-cell-type" tabindex="-1"><a class="header-anchor" href="#extract-data-for-the-current-cell-type"><span>Extract data for the current cell type</span></a></h1><p>ct.sub &lt;- subset(p65, subset = my.cell.type == celltypes[i])</p><h1 id="get-the-scaled-data-matrix" tabindex="-1"><a class="header-anchor" href="#get-the-scaled-data-matrix"><span>Get the scaled data matrix</span></a></h1><p>scaledata.mat &lt;- ct.sub@assays$RNA@scale.data</p><h1 id="calculate-the-centroid-for-the-current-cell-type-i-e-the-mean-per-row" tabindex="-1"><a class="header-anchor" href="#calculate-the-centroid-for-the-current-cell-type-i-e-the-mean-per-row"><span>Calculate the centroid for the current cell type (i.e., the mean per row)</span></a></h1><p>centroid.mat[i, ] &lt;- rowMeans(scaledata.mat) }</p><h1 id="calculate-the-correlation-matrix-of-the-centroid-matrix" tabindex="-1"><a class="header-anchor" href="#calculate-the-correlation-matrix-of-the-centroid-matrix"><span>Calculate the correlation matrix of the centroid matrix</span></a></h1><p>p65.corrmat &lt;- cor(t(centroid.mat))</p><h1 id="set-the-row-and-column-names-of-the-correlation-matrix" tabindex="-1"><a class="header-anchor" href="#set-the-row-and-column-names-of-the-correlation-matrix"><span>Set the row and column names of the correlation matrix</span></a></h1><p>rownames(p65.corrmat) &lt;- celltypes colnames(p65.corrmat) &lt;- celltypes</p><h1 id="draw-the-heatmap-without-row-and-column-clustering" tabindex="-1"><a class="header-anchor" href="#draw-the-heatmap-without-row-and-column-clustering"><span>Draw the heatmap without row and column clustering</span></a></h1><p>pheatmap(p65.corrmat, cluster_rows = F, cluster_cols = F, border_color = NA)</p><h1 id="scale-the-correlation-matrix" tabindex="-1"><a class="header-anchor" href="#scale-the-correlation-matrix"><span>Scale the correlation matrix</span></a></h1><p>scale.max &lt;- 0.5 scale.min &lt;- -0.2 scale.corrmat &lt;- p65.corrmat</p><h1 id="set-values-greater-than-scale-max-to-scale-max" tabindex="-1"><a class="header-anchor" href="#set-values-greater-than-scale-max-to-scale-max"><span>Set values greater than scale.max to scale.max</span></a></h1><p>scale.corrmat[scale.corrmat &gt; scale.max] &lt;- scale.max</p><h1 id="set-values-less-than-scale-min-to-scale-min" tabindex="-1"><a class="header-anchor" href="#set-values-less-than-scale-min-to-scale-min"><span>Set values less than scale.min to scale.min</span></a></h1><p>scale.corrmat[scale.corrmat &lt; scale.min] &lt;- scale.min</p><h1 id="draw-the-scaled-heatmap" tabindex="-1"><a class="header-anchor" href="#draw-the-scaled-heatmap"><span>Draw the scaled heatmap</span></a></h1><p>pheatmap(scale.corrmat, cluster_rows = F, cluster_cols = F, border_color = NA)</p><h1 id="this-code-block-is-used-to-generate-the-expression-of-regional-marker-genes-at-each-age" tabindex="-1"><a class="header-anchor" href="#this-code-block-is-used-to-generate-the-expression-of-regional-marker-genes-at-each-age"><span>This code block is used to generate the expression of regional marker genes at each age.</span></a></h1><h1 id="this-part-of-the-code-generates-extended-data-fig-3j-k" tabindex="-1"><a class="header-anchor" href="#this-part-of-the-code-generates-extended-data-fig-3j-k"><span>This part of the code generates Extended Data Fig. 3j - k.</span></a></h1><h1 id="first-create-a-function-for-plotting-heatmaps" tabindex="-1"><a class="header-anchor" href="#first-create-a-function-for-plotting-heatmaps"><span>First, create a function for plotting heatmaps.</span></a></h1><p>avg.gene.heatmap &lt;- function(obj, br.or.ct, # &quot;br&quot; for operation by brain region, &quot;ct&quot; for operation by cell type identity.rows, # Cell type or brain region names in order gene.columns, # Gene names in order scale.max = 2, # Maximum value for scaling heatmap data scale.min = 0, # Minimum value for scaling heatmap data return.matrix = 0) { # Whether to return the heatmap matrix, 0 for no, 1 for yes</p><h1 id="set-the-default-assay-to-rna-1" tabindex="-1"><a class="header-anchor" href="#set-the-default-assay-to-rna-1"><span>Set the default assay to RNA</span></a></h1><p>DefaultAssay(obj) &lt;- &quot;RNA&quot;</p><h1 id="set-all-genes-as-variable-features-for-subsequent-scaling" tabindex="-1"><a class="header-anchor" href="#set-all-genes-as-variable-features-for-subsequent-scaling"><span>Set all genes as variable features for subsequent scaling</span></a></h1><p>VariableFeatures(obj) &lt;- rownames(obj)</p><h1 id="scale-the-data-1" tabindex="-1"><a class="header-anchor" href="#scale-the-data-1"><span>Scale the data</span></a></h1><p>obj &lt;- ScaleData(obj)</p><h1 id="initialize-an-empty-matrix-to-store-the-heatmap-data" tabindex="-1"><a class="header-anchor" href="#initialize-an-empty-matrix-to-store-the-heatmap-data"><span>Initialize an empty matrix to store the heatmap data</span></a></h1><p>heatmap.to.plot &lt;- matrix(, nrow = length(identity.rows), ncol = length(gene.columns))</p><h1 id="iterate-over-each-row-cell-type-or-brain-region" tabindex="-1"><a class="header-anchor" href="#iterate-over-each-row-cell-type-or-brain-region"><span>Iterate over each row (cell type or brain region)</span></a></h1><p>for (ident.row in 1:length(identity.rows)) { if (br.or.ct == &quot;br&quot;) { # If operating by brain region, extract data for the corresponding brain region ident.sub &lt;- subset(obj, subset = brain.region == identity.rows[ident.row]) } if (br.or.ct == &quot;ct&quot;) { # If operating by cell type, extract data for the corresponding cell type ident.sub &lt;- subset(obj, subset = my.cell.type == identity.rows[ident.row]) }</p><pre><code># Iterate over each column (gene)
for (gene in 1:length(gene.columns)) {
  # Find the index of the current gene in the data
  gene.ind &lt;- grep(paste0(&quot;^&quot;, gene.columns[gene], &quot;$&quot;), rownames(ident.sub))

  # Calculate the average expression value of the current gene in the current cell type or brain region and store it in the heatmap matrix
  heatmap.to.plot[ident.row, gene] &lt;- mean(ident.sub@assays$RNA@scale.data[gene.ind, ])
}
</code></pre><p>}</p><h1 id="set-the-row-and-column-names-of-the-heatmap-matrix" tabindex="-1"><a class="header-anchor" href="#set-the-row-and-column-names-of-the-heatmap-matrix"><span>Set the row and column names of the heatmap matrix</span></a></h1><p>rownames(heatmap.to.plot) &lt;- identity.rows colnames(heatmap.to.plot) &lt;- gene.columns</p><h1 id="scale-the-heatmap-matrix" tabindex="-1"><a class="header-anchor" href="#scale-the-heatmap-matrix"><span>Scale the heatmap matrix</span></a></h1><p>heatmap.to.plot.scaled &lt;- heatmap.to.plot heatmap.to.plot.scaled[heatmap.to.plot.scaled &gt; scale.max] &lt;- scale.max heatmap.to.plot.scaled[heatmap.to.plot.scaled &lt; scale.min] &lt;- scale.min</p><h1 id="draw-the-heatmap-without-row-and-column-clustering-using-a-reversed-color-palette" tabindex="-1"><a class="header-anchor" href="#draw-the-heatmap-without-row-and-column-clustering-using-a-reversed-color-palette"><span>Draw the heatmap without row and column clustering, using a reversed color palette</span></a></h1><p>pheatmap(t(heatmap.to.plot.scaled), cluster_rows = F, cluster_cols = F, color = rev(bathy_deep_pal(50)), border_color = NA)</p><h1 id="if-return-matrix-is-1-return-the-heatmap-matrix" tabindex="-1"><a class="header-anchor" href="#if-return-matrix-is-1-return-the-heatmap-matrix"><span>If return.matrix is 1, return the heatmap matrix</span></a></h1><p>if (return.matrix == 1) { return(heatmap.to.plot) } }</p><h1 id="now-get-the-marker-genes-and-plot-the-heatmap" tabindex="-1"><a class="header-anchor" href="#now-get-the-marker-genes-and-plot-the-heatmap"><span>Now get the marker genes and plot the heatmap.</span></a></h1><h1 id="convert-the-brain-regions-in-the-merged-data-to-factor-type-and-specify-the-level-order" tabindex="-1"><a class="header-anchor" href="#convert-the-brain-regions-in-the-merged-data-to-factor-type-and-specify-the-level-order"><span>Convert the brain regions in the merged data to factor type and specify the level order.</span></a></h1><p>excit.merge$brain.region &lt;- factor(excit.merge$brain.region, levels = c(&quot;BNST&quot;, &quot;HDB.VLPO&quot;, &quot;LPO.PS&quot;, &quot;BAC-like&quot;, &quot;PeFA&quot;, &quot;MPN.anterior&quot;, &quot;MPN.posterior&quot;, &quot;AvPE.MnPO&quot;, &quot;PVN&quot;, &quot;PVT&quot;, &quot;Mixed/Unknown&quot;))</p><h1 id="set-the-identity-to-brain-region-for-subsequent-analysis" tabindex="-1"><a class="header-anchor" href="#set-the-identity-to-brain-region-for-subsequent-analysis"><span>Set the identity to brain region for subsequent analysis.</span></a></h1><p>Idents(excit.merge) &lt;- &quot;brain.region&quot;</p><h1 id="data-preparation" tabindex="-1"><a class="header-anchor" href="#data-preparation"><span>Data preparation</span></a></h1><h1 id="extract-data-for-age-p65" tabindex="-1"><a class="header-anchor" href="#extract-data-for-age-p65"><span>Extract data for age P65</span></a></h1><p>excit.p65 &lt;- subset(excit.merge, subset = age == &quot;p65&quot;)</p><h1 id="extract-data-for-age-e16" tabindex="-1"><a class="header-anchor" href="#extract-data-for-age-e16"><span>Extract data for age E16</span></a></h1><p>excit.e16 &lt;- subset(excit.merge, subset = age == &quot;e16&quot;)</p><h1 id="get-the-names-of-all-brain-regions" tabindex="-1"><a class="header-anchor" href="#get-the-names-of-all-brain-regions"><span>Get the names of all brain regions</span></a></h1><p>all.br &lt;- names(table(excit.p65$brain.region))</p><h1 id="get-regional-marker-genes-for-age-p65" tabindex="-1"><a class="header-anchor" href="#get-regional-marker-genes-for-age-p65"><span>Get regional marker genes for age P65</span></a></h1><p>region.markers.p65 &lt;- FindAllMarkers(excit.p65, assay = &quot;RNA&quot;, only.pos = T)</p><h1 id="filter-marker-genes-with-adjusted-p-value-less-than-0-05" tabindex="-1"><a class="header-anchor" href="#filter-marker-genes-with-adjusted-p-value-less-than-0-05"><span>Filter marker genes with adjusted p-value less than 0.05</span></a></h1><p>region.markers.p65 &lt;- region.markers.p65[region.markers.p65$p_val_adj &lt; 0.05, ]</p><h1 id="filter-marker-genes-with-average-log2-fold-change-greater-than-0-5" tabindex="-1"><a class="header-anchor" href="#filter-marker-genes-with-average-log2-fold-change-greater-than-0-5"><span>Filter marker genes with average log2 fold change greater than 0.5</span></a></h1><p>region.markers.p65 &lt;- region.markers.p65[region.markers.p65$avg_log2FC &gt; 0.5, ]</p><h1 id="remove-duplicate-marker-genes" tabindex="-1"><a class="header-anchor" href="#remove-duplicate-marker-genes"><span>Remove duplicate marker genes</span></a></h1><p>region.markers.p65 &lt;- region.markers.p65$gene[-which(duplicated(region.markers.p65$gene))]</p><h1 id="display-the-expression-of-p65-marker-genes-at-p65-age-in-a-heatmap" tabindex="-1"><a class="header-anchor" href="#display-the-expression-of-p65-marker-genes-at-p65-age-in-a-heatmap"><span>Display the expression of P65 marker genes at P65 age in a heatmap.</span></a></h1><p>avg.gene.heatmap(excit.p65, &quot;br&quot;, all.br, region.markers.p65)</p><h1 id="now-test-the-expression-of-p65-genes-at-e16-age" tabindex="-1"><a class="header-anchor" href="#now-test-the-expression-of-p65-genes-at-e16-age"><span>Now test the expression of P65 genes at E16 age.</span></a></h1><p>avg.gene.heatmap(excit.e16, &quot;br&quot;, all.br, region.markers.p65)</p><h1 id="conversely-get-regional-marker-genes-for-age-e16" tabindex="-1"><a class="header-anchor" href="#conversely-get-regional-marker-genes-for-age-e16"><span>Conversely, get regional marker genes for age E16.</span></a></h1><p>region.markers.e16 &lt;- FindAllMarkers(excit.e16, assay = &quot;RNA&quot;, only.pos = T)</p><h1 id="filter-marker-genes-with-adjusted-p-value-less-than-0-05-1" tabindex="-1"><a class="header-anchor" href="#filter-marker-genes-with-adjusted-p-value-less-than-0-05-1"><span>Filter marker genes with adjusted p-value less than 0.05</span></a></h1><p>region.markers.e16 &lt;- region.markers.e16[region.markers.e16$p_val_adj &lt; 0.05, ]</p><h1 id="filter-marker-genes-with-average-log2-fold-change-greater-than-0-5-1" tabindex="-1"><a class="header-anchor" href="#filter-marker-genes-with-average-log2-fold-change-greater-than-0-5-1"><span>Filter marker genes with average log2 fold change greater than 0.5</span></a></h1><p>region.markers.e16 &lt;- region.markers.e16[region.markers.e16$avg_log2FC &gt; 0.5, ]</p><h1 id="remove-duplicate-marker-genes-1" tabindex="-1"><a class="header-anchor" href="#remove-duplicate-marker-genes-1"><span>Remove duplicate marker genes</span></a></h1><p>region.markers.e16 &lt;- region.markers.e16$gene[-which(duplicated(region.markers.e16$gene))]</p><h1 id="display-the-expression-of-e16-marker-genes-at-e16-age-in-a-heatmap" tabindex="-1"><a class="header-anchor" href="#display-the-expression-of-e16-marker-genes-at-e16-age-in-a-heatmap"><span>Display the expression of E16 marker genes at E16 age in a heatmap.</span></a></h1><p>avg.gene.heatmap(excit.e16, &quot;br&quot;, all.br, region.markers.e16)</p><h1 id="now-test-the-expression-of-e16-genes-at-p65-age" tabindex="-1"><a class="header-anchor" href="#now-test-the-expression-of-e16-genes-at-p65-age"><span>Now test the expression of E16 genes at P65 age.</span></a></h1><p>avg.gene.heatmap(excit.p65, &quot;br&quot;, all.br, region.markers.e16)</p><h1 id="now-remove-marker-genes-that-appear-in-both-lists-and-replot-the-heatmaps" tabindex="-1"><a class="header-anchor" href="#now-remove-marker-genes-that-appear-in-both-lists-and-replot-the-heatmaps"><span>Now remove marker genes that appear in both lists and replot the heatmaps.</span></a></h1><h1 id="get-marker-genes-unique-to-p65" tabindex="-1"><a class="header-anchor" href="#get-marker-genes-unique-to-p65"><span>Get marker genes unique to P65</span></a></h1><p>unique.p65.markers &lt;- setdiff(region.markers.p65, region.markers.e16)</p><h1 id="get-marker-genes-unique-to-e16" tabindex="-1"><a class="header-anchor" href="#get-marker-genes-unique-to-e16"><span>Get marker genes unique to E16</span></a></h1><p>unique.e16.markers &lt;- setdiff(region.markers.e16, region.markers.p65)</p><h1 id="plot-the-relevant-heatmaps" tabindex="-1"><a class="header-anchor" href="#plot-the-relevant-heatmaps"><span>Plot the relevant heatmaps.</span></a></h1><p>avg.gene.heatmap(excit.p65, &quot;br&quot;, all.br, unique.p65.markers) avg.gene.heatmap(excit.e16, &quot;br&quot;, all.br, unique.p65.markers) avg.gene.heatmap(excit.e16, &quot;br&quot;, all.br, unique.e16.markers) avg.gene.heatmap(excit.p65, &quot;br&quot;, all.br, unique.e16.markers)</p><h1 id="now-explore-at-what-age-does-the-marker-gene-expression-pattern-begin-to-resemble-adulthood" tabindex="-1"><a class="header-anchor" href="#now-explore-at-what-age-does-the-marker-gene-expression-pattern-begin-to-resemble-adulthood"><span>Now, explore: At what age does the marker gene expression pattern begin to resemble adulthood?</span></a></h1><h1 id="simply-analyze-by-matrix-subtraction" tabindex="-1"><a class="header-anchor" href="#simply-analyze-by-matrix-subtraction"><span>Simply analyze by matrix subtraction.</span></a></h1><h1 id="extract-data-for-different-ages" tabindex="-1"><a class="header-anchor" href="#extract-data-for-different-ages"><span>Extract data for different ages</span></a></h1><p>excit.p28 &lt;- subset(excit.merge.noNA, subset = age == &quot;p28&quot;) excit.p18 &lt;- subset(excit.merge.noNA, subset = age == &quot;p18&quot;) excit.p10 &lt;- subset(excit.merge.noNA, subset = age == &quot;p10&quot;) excit.p4 &lt;- subset(excit.merge.noNA, subset = age == &quot;p4&quot;) excit.p0 &lt;- subset(excit.merge.noNA, subset = age == &quot;p0&quot;) excit.e18 &lt;- subset(excit.merge.noNA, subset = age == &quot;e18&quot;)</p><h1 id="for-p65-marker-genes-excluding-e16-marker-genes" tabindex="-1"><a class="header-anchor" href="#for-p65-marker-genes-excluding-e16-marker-genes"><span>For P65 marker genes (excluding E16 marker genes)</span></a></h1><h1 id="get-the-expression-matrix-of-these-marker-genes-at-different-ages" tabindex="-1"><a class="header-anchor" href="#get-the-expression-matrix-of-these-marker-genes-at-different-ages"><span>Get the expression matrix of these marker genes at different ages</span></a></h1><p>p65.mat &lt;- avg.gene.heatmap(excit.p65, &quot;br&quot;, all.br, unique.p65.markers, return.matrix = 1) p28.mat &lt;- avg.gene.heatmap(excit.p28, &quot;br&quot;, all.br, unique.p65.markers, return.matrix = 1) p18.mat &lt;- avg.gene.heatmap(excit.p18, &quot;br&quot;, all.br, unique.p65.markers, return.matrix = 1) p10.mat &lt;- avg.gene.heatmap(excit.p10, &quot;br&quot;, all.br, unique.p65.markers, return.matrix = 1) p4.mat &lt;- avg.gene.heatmap(excit.p4, &quot;br&quot;, all.br, unique.p65.markers, return.matrix = 1) p0.mat &lt;- avg.gene.heatmap(excit.p0, &quot;br&quot;, all.br, unique.p65.markers, return.matrix = 1) e18.mat &lt;- avg.gene.heatmap(excit.e18, &quot;br&quot;, all.br, unique.p65.markers, return.matrix = 1) e16.mat &lt;- avg.gene.heatmap(excit.e16, &quot;br&quot;, all.br, unique.p65.markers, return.matrix = 1)</p><h1 id="calculate-the-correlation-between-the-expression-matrix-at-different-ages-and-the-expression-matrix-at-p65-age" tabindex="-1"><a class="header-anchor" href="#calculate-the-correlation-between-the-expression-matrix-at-different-ages-and-the-expression-matrix-at-p65-age"><span>Calculate the correlation between the expression matrix at different ages and the expression matrix at P65 age.</span></a></h1><p>cor.to.p65 &lt;- data.frame(age = 1:8, cor = c(cor(c(e16.mat), c(p65.mat)), cor(c(e18.mat), c(p65.mat)), cor(c(p0.mat), c(p65.mat)), cor(c(p4.mat), c(p65.mat)), cor(c(p10.mat), c(p65.mat)), cor(c(p18.mat), c(p65.mat)), cor(c(p28.mat), c(p65.mat)), 1 ))</p><h1 id="draw-a-line-plot-of-correlation-changing-with-age" tabindex="-1"><a class="header-anchor" href="#draw-a-line-plot-of-correlation-changing-with-age"><span>Draw a line plot of correlation changing with age.</span></a></h1><p>ggplot(cor.to.p65, aes(x = age, y = cor)) + geom_line(aes(linewidth = 4)) + theme_classic() + ylim(0.3, 1) + theme(text = element_text(family = &quot;Myriad Pro&quot;)) + theme(text = element_text(size = 20))</p></div></div></div></div></div><footer class="lp-footer-wrapper" vp-footer data-v-a3194778 data-v-27bdb921><div class="lp-footer" data-v-27bdb921>Default footer<br data-v-27bdb921>Running time: 570 days 18 hours 28 minutes 27 seconds</div><div class="lp-copyright" data-v-27bdb921>Copyright © 2024 Miss Li </div></footer><!--]--></div><!--[--><!----><!--]--><!--]--></div>
    <script type="module" src="/assets/app-CrqbF7DL.js" defer></script>
  </body>
</html>
