import{_ as a,e as i,g as t,o as s}from"./app-DJxBf6IT.js";const n="/assets/eng01-DRlxvTIZ.png",o={};function r(l,e){return s(),i("div",null,e[0]||(e[0]=[t('<h2 id="research-description" tabindex="-1"><a class="header-anchor" href="#research-description"><span><span style="color:#2ea760ff;">Research Description:</span></span></a></h2><p>• Preprocessed and clustered scRNA-seq data (16 PDAC tumors, 3 normal tissues; 16 PDAC PBMCs, 4 healthy PBMCs) using Harmony for batch correction and SingleR/Seurat v5 for annotation, to identify spatial collapse of T cells and TAM expansion associated with COL1A1-driven ECM remodeling.</p><p>• Conducted pseudotime trajectory inference on myeloid-to-macrophage lineages via Monocle3, uncovering epigenetic silencing of cytotoxic pathways (GZMA/NKG7 downregulation) and upregulation of ribosomal biogenesis in PDAC-derived TAMs.</p><p>• Processed and analyzed 10x Xenium FFPE spatial transcriptomic data using Squidpy + Scanpy, followed by DESeq2 and FindAllMarkers for DEG analysis, and GO/KEGG enrichment to identify key immune modulators.</p><p>• Identified bifunctional regulatory molecules including NOP53 (activates p53, suppresses PI3K-AKT) and SPP1 (involved in both metastasis and anti-tumor immunity), and validated their spatial co-localization with TAMs and tumor cells.</p><h2 id="research-background-and-objectives" tabindex="-1"><a class="header-anchor" href="#research-background-and-objectives"><span><span style="color:#2ea760ff;">Research Background and Objectives:</span></span></a></h2><p>Pancreatic ductal adenocarcinoma (PDAC) is a highly lethal cancer with limited early diagnostic and therapeutic options. This study integrates single-cell and spatial transcriptomics to dissect the spatiotemporal heterogeneity of tumor-associated macrophages (TAMs) and their dual roles in immune suppression and tumor progression, aiming to identify novel biomarkers and therapeutic targets.</p><h2 id="methods" tabindex="-1"><a class="header-anchor" href="#methods"><span><span style="color:#2ea760ff;">Methods:</span></span></a></h2><p>1.Utilized publicly available scRNA-seq and spatial transcriptomic data from GEO and 10x Genomics.</p><p>2.Integrated datasets and annotated cell types using Seurat, Harmony, and SingleR.</p><p>3.Conducted pseudotemporal trajectory analysis with Monocle3 to map macrophage differentiation.</p><p>4.Performed differential expression analysis with DESeq2 and pathway enrichment with GO/KEGG.</p><p>5.Visualized spatial gene expression patterns using Squidpy. <img src="'+n+'" alt="pic2"></p><h2 id="key-findings" tabindex="-1"><a class="header-anchor" href="#key-findings"><span><span style="color:#2ea760ff;">Key Findings:</span></span></a></h2><p>1.T-cells showed restricted spatial distribution in PDAC, while macrophages expanded, indicating impaired immune surveillance.</p><p>2.Pseudotime analysis revealed myeloid progenitor differentiation into monocytes/macrophages, with cytotoxic pathways suppressed in PDAC TAMs.</p><p>3.Identified tissue-specific markers: LCN2 in healthy tissues, CTRB1/AMY2A in PDAC.</p><p>4.Discovered dual-function molecular hubs: NOP53 (inhibits PI3K-AKT, activates p53) and SPP1 (pro-metastatic yet immunostimulatory).</p><p>5.Spatial transcriptomics highlighted NOP53, CFB, and SPP1 as potential diagnostic biomarkers.</p>',19)]))}const d=a(o,[["render",r],["__file","article-1.html.vue"]]),p=JSON.parse('{"path":"/research/article-1.html","title":"Spatiotemporal Transcriptomic Analysis of Tumor-Associated Macrophage Heterogeneity and Bifunctional Molecular Nodes in Pancreatic Ductal Adenocarcinoma","lang":"en-US","frontmatter":{"date":"06/2024-04/2025","layout":"ResearchPost","sort":0,"category":[1,"Research Assistant (remote)"],"title":"Spatiotemporal Transcriptomic Analysis of Tumor-Associated Macrophage Heterogeneity and Bifunctional Molecular Nodes in Pancreatic Ductal Adenocarcinoma","teacher":"Prof. Guangwei Zhang from USC"},"headers":[{"level":2,"title":"Research Description:","slug":"research-description","link":"#research-description","children":[]},{"level":2,"title":"Research Background and Objectives:","slug":"research-background-and-objectives","link":"#research-background-and-objectives","children":[]},{"level":2,"title":"Methods:","slug":"methods","link":"#methods","children":[]},{"level":2,"title":"Key Findings:","slug":"key-findings","link":"#key-findings","children":[]}],"git":{"updatedTime":1756120529000,"contributors":[{"name":"limengli","username":"limengli","email":"limengli@qq.com","commits":4,"url":"https://github.com/limengli"}],"changelog":[{"hash":"8aa5b7213f2720f2ad94bd1147f439d5f78fcb8a","time":1756120529000,"email":"limengli@qq.com","author":"limengli","message":"update"},{"hash":"055a3898aed9637227afe2ba0e8f86051432838c","time":1756012197000,"email":"limengli@qq.com","author":"limengli","message":"update"},{"hash":"84a449896542c2afd785917d68f97d41e64b66fb","time":1756009719000,"email":"limengli@qq.com","author":"limengli","message":"update"},{"hash":"33fb064215db0208e223b56976bfe0fd34434077","time":1754478598000,"email":"limengli@qq.com","author":"limengli","message":"web"}]},"filePathRelative":"research/article-1.md","excerpt":"<h2><span style=\\"color: #2ea760ff;\\">Research Description:</span></h2>\\n<p>• Preprocessed and clustered scRNA-seq data (16 PDAC tumors, 3 normal tissues; 16 PDAC PBMCs, 4 healthy PBMCs) using Harmony for batch correction and SingleR/Seurat v5 for annotation, to identify spatial collapse of T cells and TAM expansion associated with COL1A1-driven ECM remodeling.</p>"}');export{d as comp,p as data};
