import{_ as n,e as t,f as e,o as r}from"./app-OXf-X-ix.js";const o={};function a(s,i){return r(),t("div",null,i[0]||(i[0]=[e("h2",{id:"projectdescription",tabindex:"-1"},[e("a",{class:"header-anchor",href:"#projectdescription"},[e("span",null,"projectDescription:")])],-1),e("p",null,"• Led a full-stack genomic and structural bioinformatics analysis of HMTp210, a hemagglutinin protein from Avibacterium paragallinarum (Apg), and pioneered the first sequence-based serotyping model for Apg using 91 strains from NCBI. • Constructed a robust serotyping framework integrating phylogenetic tree building (PhyloPhlAn), pan-genome analysis (BPGA + MAFFT), and motif detection (MEME), classifying HMTp210 into 3 major serogroups (A/B/C) and 8 subtypes (A1–C3) based on Region2-specific divergence. • Applied AlphaFold2 for full-length 3D structure prediction of A/B/C representatives, revealing HMTp210 as a trimeric TAA (trimeric autotransporter adhesin) with well-defined head (β-sheet), stalk (“α-helix–loop–β”) and anchor (β-barrel) domains. • Performed B-cell epitope prediction using IEDB and Rankpep, identifying 6 conserved immunodominant epitopes enriched in Region2/3, including a novel “clover-shaped loop” structure with >95% cross-serotype conservation. • Identified A-type specific epitope (1634N–1640T) and C2-subtype–unique insertions, enabling development of a motif-based subtyping model and offering theoretical groundwork for subtype-specific recombinant vaccine design.",-1)]))}const c=n(o,[["render",a],["__file","article-2.html.vue"]]),p=JSON.parse('{"path":"/research/article-2.html","title":"Bioinformatics Research on HMTp210-Based Serotyping and Epitope Profiling of Avibacterium Paragallinarum","lang":"en-US","frontmatter":{"date":"12/2024-03/2025","layout":"ResearchPost","sort":0,"category":[1,"Research Assistant"],"title":"Bioinformatics Research on HMTp210-Based Serotyping and Epitope Profiling of Avibacterium Paragallinarum","teacher":"Supervised by Prof. Min Chen from Shandong University"},"headers":[{"level":2,"title":"projectDescription:","slug":"projectdescription","link":"#projectdescription","children":[]}],"git":{"updatedTime":1754478598000,"contributors":[{"name":"limengli","username":"limengli","email":"limengli@qq.com","commits":1,"url":"https://github.com/limengli"}],"changelog":[{"hash":"33fb064215db0208e223b56976bfe0fd34434077","time":1754478598000,"email":"limengli@qq.com","author":"limengli","message":"web"}]},"filePathRelative":"research/article-2.md","excerpt":"<h2>projectDescription:</h2>\\n<p>• Led a full-stack genomic and structural bioinformatics analysis of HMTp210, a hemagglutinin protein from Avibacterium paragallinarum (Apg), and pioneered the first sequence-based serotyping model for Apg using 91 strains from NCBI.\\n• Constructed a robust serotyping framework integrating phylogenetic tree building (PhyloPhlAn), pan-genome analysis (BPGA + MAFFT), and motif detection (MEME), classifying HMTp210 into 3 major serogroups (A/B/C) and 8 subtypes (A1–C3) based on Region2-specific divergence.\\n• Applied AlphaFold2 for full-length 3D structure prediction of A/B/C representatives, revealing HMTp210 as a trimeric TAA (trimeric autotransporter adhesin) with well-defined head (β-sheet), stalk (“α-helix–loop–β”) and anchor (β-barrel) domains.\\n• Performed B-cell epitope prediction using IEDB and Rankpep, identifying 6 conserved immunodominant epitopes enriched in Region2/3, including a novel “clover-shaped loop” structure with &gt;95% cross-serotype conservation.\\n• Identified A-type specific epitope (1634N–1640T) and C2-subtype–unique insertions, enabling development of a motif-based subtyping model and offering theoretical groundwork for subtype-specific recombinant vaccine design.</p>"}');export{c as comp,p as data};
