import{_ as i,e as n,g as a,o as t}from"./app-BY3UUy9i.js";const s="/assets/eng02-SR_0TuUz.jpg",o={};function r(c,e){return t(),n("div",null,e[0]||(e[0]=[a('<h2 id="research-description" tabindex="-1"><a class="header-anchor" href="#research-description"><span><span style="color:#2ea760ff;">Research Description:</span></span></a></h2><p>• Led a full-stack genomic and structural bioinformatics analysis of HMTp210, a hemagglutinin protein from Avibacterium paragallinarum (Apg), and pioneered the first sequence-based serotyping model for Apg using 91 strains from NCBI.</p><p>• Constructed a robust serotyping framework integrating phylogenetic tree building (PhyloPhlAn), pan-genome analysis (BPGA + MAFFT), and motif detection (MEME), classifying HMTp210 into 3 major serogroups (A/B/C) and 8 subtypes (A1–C3) based on Region2-specific divergence.</p><p>• Applied AlphaFold2 for full-length 3D structure prediction of A/B/C representatives, revealing HMTp210 as a trimeric TAA (trimeric autotransporter adhesin) with well-defined head (β-sheet), stalk (“α-helix–loop–β”) and anchor (β-barrel) domains.</p><p>• Performed B-cell epitope prediction using IEDB and Rankpep, identifying 6 conserved immunodominant epitopes enriched in Region2/3, including a novel “clover-shaped loop” structure with &gt;95% cross-serotype conservation.</p><p>• Identified A-type specific epitope (1634N–1640T) and C2-subtype–unique insertions, enabling development of a motif-based subtyping model and offering theoretical groundwork for subtype-specific recombinant vaccine design.</p><h2 id="research-background-and-objectives" tabindex="-1"><a class="header-anchor" href="#research-background-and-objectives"><span><span style="color:#2ea760ff;">Research Background and Objectives:</span></span></a></h2><p>Avibacterium paragallinarum (Apg) is the causative agent of Infectious Coryza (IC) in chickens, leading to significant economic losses in poultry farming. Its hemagglutinin protein HMTp210, located on the outer membrane, plays a crucial role in immunogenicity and pathogenicity, can stimulate host immune responses, and is a potential subunit vaccine antigen. However, systematic research on HMTp210 is limited. This study aims to analyze the differences in HMTp210 across different serotypes from genomic and structural perspectives, exploring its potential as a basis for serotyping and a target for vaccine development.</p><h2 id="methods" tabindex="-1"><a class="header-anchor" href="#methods"><span><span style="color:#2ea760ff;">Methods:</span></span></a></h2><p>1.Acquired genomic and HMTp210 sequence data of various Apg serotypes from NCBI.</p><p>2.Constructed phylogenetic trees using PhylPHLAN and pan-genomes using BPGA to analyze evolutionary relationships.</p><p>3.Performed gene homology, motif, and sequence conservation analyses using SimPlot, Jalview, MEME, and Weblogo.</p><p>4.Developed machine learning models (SVM, etc.) to predict serotypes of unknown strains based on the hydrophobicity of amino acids in HMTp210 Region 2.</p><p>5.Predicted B-cell and MHC-binding epitopes using IEDB and Rankpep.</p><p>6.Predicted the 3D structure of HMTp210 for three representative serotype strains using AlphaFold2. <img src="'+s+'" alt="pic2"></p><h2 id="key-findings" tabindex="-1"><a class="header-anchor" href="#key-findings"><span><span style="color:#2ea760ff;">Key Findings:</span></span></a></h2><p>1.Phylogenetic analysis based on HMTp210 clearly classified Apg into three serogroups (A, B, C) with internal subgroups (e.g., A1, A2, A3; B1, B2; C1, C2, C3).</p><p>2.The hmtp210 gene could be divided into three regions (Region1, 2, 3), with Region2 exhibiting the highest sequence divergence, making it the focus for serotype difference analysis.</p><p>3.A machine learning model (SVM) achieved ~98% accuracy in predicting serotypes based on the hydrophobicity of 267 amino acid positions in Region 2.</p><p>4.Epitope predictions (B-cell and MHC-binding) were concentrated in Region 2 and Region 3.</p><p>5.AlphaFold2 structure prediction confirmed HMTp210 is a classic trimeric autotransporter adhesin (TAA family protein). A structurally unique &quot;Clover Loop&quot; region was identified within the predicted epitopes, adjacent to a highly conserved area. This structure may influence immune recognition and pathogen colonization.</p><p>6.A unique insertion sequence was identified in the C2 subgroup.</p>',22)]))}const l=i(o,[["render",r],["__file","article-2.html.vue"]]),d=JSON.parse('{"path":"/research/article-2.html","title":"Bioinformatics Research on HMTp210-Based Serotyping and Epitope Profiling of Avibacterium Paragallinarum","lang":"en-US","frontmatter":{"date":"12/2024-03/2025","layout":"ResearchPost","sort":0,"category":[1,"Research Assistant"],"title":"Bioinformatics Research on HMTp210-Based Serotyping and Epitope Profiling of Avibacterium Paragallinarum","teacher":"Supervised by Prof. Min Chen from Shandong University"},"headers":[{"level":2,"title":"Research Description:","slug":"research-description","link":"#research-description","children":[]},{"level":2,"title":"Research Background and Objectives:","slug":"research-background-and-objectives","link":"#research-background-and-objectives","children":[]},{"level":2,"title":"Methods:","slug":"methods","link":"#methods","children":[]},{"level":2,"title":"Key Findings:","slug":"key-findings","link":"#key-findings","children":[]}],"git":{"updatedTime":1756012197000,"contributors":[{"name":"limengli","username":"limengli","email":"limengli@qq.com","commits":3,"url":"https://github.com/limengli"}],"changelog":[{"hash":"055a3898aed9637227afe2ba0e8f86051432838c","time":1756012197000,"email":"limengli@qq.com","author":"limengli","message":"update"},{"hash":"84a449896542c2afd785917d68f97d41e64b66fb","time":1756009719000,"email":"limengli@qq.com","author":"limengli","message":"update"},{"hash":"33fb064215db0208e223b56976bfe0fd34434077","time":1754478598000,"email":"limengli@qq.com","author":"limengli","message":"web"}]},"filePathRelative":"research/article-2.md","excerpt":"<h2><span style=\\"color: #2ea760ff;\\">Research Description:</span></h2>\\n<p>• Led a full-stack genomic and structural bioinformatics analysis of HMTp210, a hemagglutinin protein from Avibacterium paragallinarum (Apg), and pioneered the first sequence-based serotyping model for Apg using 91 strains from NCBI.</p>"}');export{l as comp,d as data};
